// @billion-business/ai-core/dist/index.js
// src/providers/gemini.ts
import { GoogleGenAI } from "@google/genai";
var GeminiProvider = class {
  constructor(apiKey) {
    this.client = null;
    this.apiKey = apiKey;
  }
  getClient() {
    if (!this.client) {
      this.client = new GoogleGenAI({ apiKey: this.apiKey });
    }
    return this.client;
  }
  async generate(prompt, options = {}) {
    const start = Date.now();
    const client = this.getClient();
    const model = options.model || "gemini-2.0-flash";
    const contents = options.systemPrompt ? `${options.systemPrompt}

${prompt}` : prompt;
    const response = await client.models.generateContent({
      model,
      contents,
      config: {
        temperature: options.temperature ?? 0.7,
        maxOutputTokens: options.maxTokens ?? 4096,
        responseMimeType: options.responseFormat === "json" ? "application/json" : void 0
      }
    });
    const text = response.text || "";
    const latencyMs = Date.now() - start;
    return {
      text,
      provider: "gemini",
      model,
      tokensUsed: response.usageMetadata?.totalTokenCount,
      costUsd: this.estimateCost(model, response.usageMetadata?.totalTokenCount || 0),
      latencyMs
    };
  }
  async chat(messages, options = {}) {
    const start = Date.now();
    const client = this.getClient();
    const model = options.model || "gemini-2.0-flash";
    const contents = messages.map((m) => ({
      role: m.role === "assistant" ? "model" : "user",
      parts: [{ text: m.content }]
    }));
    const chat = client.chats.create({
      model,
      config: {
        temperature: options.temperature ?? 0.7,
        maxOutputTokens: options.maxTokens ?? 4096
      }
    });
    const lastMessage = messages[messages.length - 1];
    const response = await chat.sendMessage({ message: lastMessage.content });
    const text = response.text || "";
    const latencyMs = Date.now() - start;
    return {
      text,
      provider: "gemini",
      model,
      tokensUsed: response.usageMetadata?.totalTokenCount,
      costUsd: this.estimateCost(model, response.usageMetadata?.totalTokenCount || 0),
      latencyMs
    };
  }
  estimateCost(model, tokens) {
    const pricePer1K = model.includes("pro") ? 35e-4 : 1e-4;
    return tokens / 1e3 * pricePer1K;
  }
};

// src/providers/openai.ts
import OpenAI from "openai";
var OpenAIProvider = class {
  constructor(apiKey, baseUrl) {
    this.client = new OpenAI({
      apiKey,
      baseURL: baseUrl,
      dangerouslyAllowBrowser: true
    });
  }
  async generate(prompt, options = {}) {
    const start = Date.now();
    const model = options.model || "gpt-4o";
    const messages = [];
    if (options.systemPrompt) {
      messages.push({ role: "system", content: options.systemPrompt });
    }
    messages.push({ role: "user", content: prompt });
    const response = await this.client.chat.completions.create({
      model,
      messages,
      temperature: options.temperature ?? 0.7,
      max_tokens: options.maxTokens ?? 4096,
      response_format: options.responseFormat === "json" ? { type: "json_object" } : void 0
    });
    const text = response.choices[0]?.message?.content || "";
    const latencyMs = Date.now() - start;
    const tokensUsed = response.usage?.total_tokens;
    return {
      text,
      provider: "openai",
      model,
      tokensUsed,
      costUsd: this.estimateCost(model, tokensUsed || 0),
      latencyMs
    };
  }
  async chat(messages, options = {}) {
    const start = Date.now();
    const model = options.model || "gpt-4o";
    const oaiMessages = messages.map((m) => ({
      role: m.role === "assistant" ? "assistant" : m.role === "system" ? "system" : "user",
      content: m.content
    }));
    const response = await this.client.chat.completions.create({
      model,
      messages: oaiMessages,
      temperature: options.temperature ?? 0.7,
      max_tokens: options.maxTokens ?? 4096
    });
    const text = response.choices[0]?.message?.content || "";
    const latencyMs = Date.now() - start;
    const tokensUsed = response.usage?.total_tokens;
    return {
      text,
      provider: "openai",
      model,
      tokensUsed,
      costUsd: this.estimateCost(model, tokensUsed || 0),
      latencyMs
    };
  }
  estimateCost(model, tokens) {
    const prices = {
      "gpt-4o": 5e-3,
      "gpt-4o-mini": 15e-5,
      "o3": 0.01,
      "o3-mini": 11e-4
    };
    const pricePer1K = prices[model] || 5e-3;
    return tokens / 1e3 * pricePer1K;
  }
};

// src/providers/deepseek.ts
var DeepSeekProvider = class {
  constructor(apiKey) {
    this.apiKey = apiKey;
    this.baseUrl = "https://api.deepseek.com/v1/chat/completions";
  }
  async generate(prompt, options = {}) {
    const start = Date.now();
    const model = options.model || "deepseek-chat";
    const messages = [];
    if (options.systemPrompt) {
      messages.push({ role: "system", content: options.systemPrompt });
    }
    messages.push({ role: "user", content: prompt });
    const response = await fetch(this.baseUrl, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${this.apiKey}`
      },
      body: JSON.stringify({
        model,
        messages,
        temperature: options.temperature ?? 0.7,
        max_tokens: options.maxTokens ?? 1024
      })
    });
    if (!response.ok) {
      const errorText = await response.text().catch(() => "No details");
      throw new Error(`DeepSeek API failed with status ${response.status}: ${errorText}`);
    }
    const data = await response.json();
    const text = data.choices?.[0]?.message?.content || "";
    const latencyMs = Date.now() - start;
    const tokensUsed = data.usage?.total_tokens;
    return {
      text,
      provider: "deepseek",
      model,
      tokensUsed,
      costUsd: this.estimateCost(tokensUsed || 0),
      latencyMs
    };
  }
  estimateCost(tokens) {
    return tokens / 1e3 * 14e-5;
  }
};

// src/AICore.ts
var DEFAULT_ROUTES = [
  { taskType: "coding", provider: "openai", model: "o3", fallback: { provider: "openai", model: "gpt-4o" } },
  { taskType: "business_logic", provider: "openai", model: "gpt-4o", fallback: { provider: "gemini", model: "gemini-2.0-flash" } },
  { taskType: "agent_brain", provider: "openai", model: "gpt-4o", fallback: { provider: "gemini", model: "gemini-2.0-flash" } },
  { taskType: "cross_domain", provider: "gemini", model: "gemini-2.5-pro", fallback: { provider: "gemini", model: "gemini-2.0-flash" } },
  { taskType: "creative", provider: "gemini", model: "gemini-2.0-flash", fallback: { provider: "deepseek", model: "deepseek-chat" } },
  { taskType: "analysis", provider: "gemini", model: "gemini-2.0-flash", fallback: { provider: "deepseek", model: "deepseek-chat" } },
  { taskType: "summarization", provider: "deepseek", model: "deepseek-chat", fallback: { provider: "gemini", model: "gemini-2.0-flash" } },
  { taskType: "chat", provider: "gemini", model: "gemini-2.0-flash", fallback: { provider: "deepseek", model: "deepseek-chat" } }
];
var AICore = class {
  constructor(config) {
    this.gemini = null;
    this.openai = null;
    this.deepseek = null;
    this.callLog = [];
    this.config = config;
    this.routes = config.routes || DEFAULT_ROUTES;
    this.stats = {
      totalCalls: 0,
      totalTokens: 0,
      totalCostUsd: 0,
      callsByProvider: {},
      callsByTask: {},
      averageLatencyMs: 0,
      errorCount: 0
    };
    if (config.providers.gemini?.apiKey) {
      this.gemini = new GeminiProvider(config.providers.gemini.apiKey);
    }
    if (config.providers.openai?.apiKey) {
      this.openai = new OpenAIProvider(config.providers.openai.apiKey, config.providers.openai.baseUrl);
    }
    if (config.providers.deepseek?.apiKey) {
      this.deepseek = new DeepSeekProvider(config.providers.deepseek.apiKey);
    }
  }
  async generate(prompt, options = {}) {
    const taskType = options.taskType || this.detectTaskType(prompt);
    const route = this.findRoute(taskType);
    if (!route) {
      throw new Error(`No route found for task type: ${taskType}`);
    }
    const oversightResult = await this.checkOversight({
      action: "generate",
      provider: route.provider,
      model: route.model,
      prompt,
      confidence: 1,
      risk: this.assessRisk(taskType),
      metadata: options.metadata
    });
    if (!oversightResult) {
      return {
        text: "Action blocked by human oversight.",
        provider: route.provider,
        model: route.model,
        latencyMs: 0,
        flagged: true,
        flagReason: "Blocked by human oversight"
      };
    }
    if (this.config.costTracking?.enabled && this.config.costTracking.budgetLimitUsd) {
      if (this.stats.totalCostUsd >= this.config.costTracking.budgetLimitUsd) {
        throw new Error(`AI budget exceeded: $${this.stats.totalCostUsd.toFixed(2)} / $${this.config.costTracking.budgetLimitUsd.toFixed(2)}`);
      }
    }
    let result = null;
    let error = null;
    try {
      result = await this.executeProvider(route.provider, route.model, prompt, options);
    } catch (e) {
      error = e;
      console.warn(`Primary provider ${route.provider} failed: ${error.message}. Trying fallback...`);
      if (route.fallback) {
        try {
          result = await this.executeProvider(route.fallback.provider, route.fallback.model, prompt, options);
        } catch (fallbackError) {
          this.stats.errorCount += 2;
          throw fallbackError;
        }
      } else {
        this.stats.errorCount++;
        throw error;
      }
    }
    if (result) {
      this.updateStats(result, taskType);
    }
    return result;
  }
  async chat(messages, options = {}) {
    const taskType = options.taskType || "chat";
    const route = this.findRoute(taskType);
    if (!route) {
      throw new Error(`No route found for task type: ${taskType}`);
    }
    let result = null;
    try {
      result = await this.executeProviderChat(route.provider, route.model, messages, options);
    } catch (e) {
      if (route.fallback) {
        result = await this.executeProviderChat(route.fallback.provider, route.fallback.model, messages, options);
      } else {
        throw e;
      }
    }
    if (result) {
      this.updateStats(result, taskType);
    }
    return result;
  }
  async executeProvider(provider, model, prompt, options) {
    switch (provider) {
      case "gemini":
        if (!this.gemini) throw new Error("Gemini provider not configured");
        return this.gemini.generate(prompt, { ...options, model });
      case "openai":
        if (!this.openai) throw new Error("OpenAI provider not configured");
        return this.openai.generate(prompt, { ...options, model });
      case "deepseek":
        if (!this.deepseek) throw new Error("DeepSeek provider not configured");
        return this.deepseek.generate(prompt, { ...options, model });
      default:
        throw new Error(`Unsupported provider: ${provider}`);
    }
  }
  async executeProviderChat(provider, model, messages, options) {
    const formattedMessages = messages.map((m) => ({
      role: m.role,
      content: m.content
    }));
    switch (provider) {
      case "gemini":
        if (!this.gemini) throw new Error("Gemini provider not configured");
        return this.gemini.chat(formattedMessages, { ...options, model });
      case "openai":
        if (!this.openai) throw new Error("OpenAI provider not configured");
        return this.openai.chat(formattedMessages, { ...options, model });
      default:
        throw new Error(`Chat not supported for provider: ${provider}`);
    }
  }
  findRoute(taskType) {
    return this.routes.find((r) => r.taskType === taskType) || null;
  }
  detectTaskType(prompt) {
    const p = prompt.toLowerCase();
    if (/\b(code|function|class|component|api|endpoint|query|schema)\b/.test(p)) return "coding";
    if (/\b(strategy|plan|analyze|forecast|budget|revenue|market)\b/.test(p)) return "business_logic";
    if (/\b(agent|autonomous|orchestrat|workflow|pipeline)\b/.test(p)) return "agent_brain";
    if (/\b(summarize|brief|overview|condense)\b/.test(p)) return "summarization";
    if (/\b(write|create|generate|story|copy|marketing)\b/.test(p)) return "creative";
    if (/\b(analyze|review|audit|evaluate|score)\b/.test(p)) return "analysis";
    return "chat";
  }
  assessRisk(taskType) {
    if (taskType === "coding" || taskType === "business_logic") return "high";
    if (taskType === "agent_brain") return "medium";
    return "low";
  }
  async checkOversight(payload) {
    const oversight = this.config.oversight;
    if (!oversight) return true;
    if (oversight.mode === "recovery") return false;
    if (oversight.mode === "shadow") {
      if (payload.confidence >= oversight.confidenceThreshold && payload.risk !== "critical") {
        return true;
      }
    }
    if (oversight.mode === "checkpoint") {
      if (oversight.onEscalate) {
        return await oversight.onEscalate(payload);
      }
    }
    return true;
  }
  updateStats(result, taskType) {
    this.stats.totalCalls++;
    this.stats.totalTokens += result.tokensUsed || 0;
    this.stats.totalCostUsd += result.costUsd || 0;
    this.stats.callsByProvider[result.provider] = (this.stats.callsByProvider[result.provider] || 0) + 1;
    this.stats.callsByTask[taskType] = (this.stats.callsByTask[taskType] || 0) + 1;
    this.callLog.push({
      timestamp: Date.now(),
      provider: result.provider,
      tokens: result.tokensUsed || 0,
      cost: result.costUsd || 0,
      latency: result.latencyMs
    });
    const totalLatency = this.callLog.reduce((sum, c) => sum + c.latency, 0);
    this.stats.averageLatencyMs = totalLatency / this.callLog.length;
  }
  getStats() {
    return { ...this.stats };
  }
  addRoute(route) {
    this.routes = this.routes.filter((r) => r.taskType !== route.taskType);
    this.routes.push(route);
  }
  setOversightMode(mode) {
    if (this.config.oversight) {
      this.config.oversight.mode = mode;
    }
  }
};

// src/middleware/oversight.ts
var OversightMiddleware = class {
  constructor(config = {}) {
    this.queue = [];
    this.history = [];
    this.mode = config.mode || "checkpoint";
    this.confidenceThreshold = config.confidenceThreshold ?? 0.7;
    this.onEscalate = config.onEscalate;
  }
  async shouldProceed(payload) {
    if (this.mode === "shadow") {
      return payload.confidence >= this.confidenceThreshold && payload.risk !== "critical";
    }
    if (this.mode === "recovery") {
      return false;
    }
    if (this.mode === "checkpoint") {
      if (this.onEscalate) {
        return await this.onEscalate(payload);
      }
      return this.autoApprove(payload);
    }
    return true;
  }
  autoApprove(payload) {
    if (payload.risk === "critical") return false;
    if (payload.confidence >= 0.9 && payload.risk === "low") return true;
    return false;
  }
  enqueue(payload) {
    const id = `ESC-${Date.now()}-${Math.random().toString(36).slice(2, 6)}`;
    this.queue.push({ ...payload, id, createdAt: /* @__PURE__ */ new Date() });
    return id;
  }
  approve(id) {
    const idx = this.queue.findIndex((q) => q.id === id);
    if (idx !== -1) {
      const item = this.queue[idx];
      this.history.push({ ...item, status: "approved", resolvedAt: /* @__PURE__ */ new Date() });
      this.queue.splice(idx, 1);
    }
  }
  reject(id) {
    const idx = this.queue.findIndex((q) => q.id === id);
    if (idx !== -1) {
      const item = this.queue[idx];
      this.history.push({ ...item, status: "rejected", resolvedAt: /* @__PURE__ */ new Date() });
      this.queue.splice(idx, 1);
    }
  }
  getQueue() {
    return [...this.queue];
  }
  getHistory() {
    return [...this.history];
  }
  setMode(mode) {
    this.mode = mode;
  }
  getMode() {
    return this.mode;
  }
  getStats() {
    return {
      pending: this.queue.length,
      approved: this.history.filter((h) => h.status === "approved").length,
      rejected: this.history.filter((h) => h.status === "rejected").length,
      mode: this.mode
    };
  }
};
export {
  AICore,
  DeepSeekProvider,
  GeminiProvider,
  OpenAIProvider,
  OversightMiddleware
};
