// @billion-business/venture-ui/dist/components/ai/AgentFleet.js
'use client';

// src/components/ai/AgentFleet.tsx
import { useState } from "react";
import { Users, Cpu, Zap, Activity, Network, Bot, Terminal as TerminalIcon, CheckCircle2, Loader2, Plus, AlertTriangle } from "lucide-react";

// src/components/ui/card.tsx
import * as React from "react";

// src/lib/utils.ts
import { clsx } from "clsx";
import { twMerge } from "tailwind-merge";
function cn(...inputs) {
  return twMerge(clsx(inputs));
}

// src/components/ui/card.tsx
import { jsx } from "react/jsx-runtime";
var Card = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      className: cn("rounded-xl border bg-card text-card-foreground shadow", className),
      ...props
    }
  )
);
Card.displayName = "Card";
var CardHeader = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("flex flex-col space-y-1.5 p-6", className), ...props })
);
CardHeader.displayName = "CardHeader";
var CardTitle = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("font-semibold leading-none tracking-tight", className), ...props })
);
CardTitle.displayName = "CardTitle";
var CardDescription = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("text-sm text-muted-foreground", className), ...props })
);
CardDescription.displayName = "CardDescription";
var CardContent = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("p-6 pt-0", className), ...props })
);
CardContent.displayName = "CardContent";
var CardFooter = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("flex items-center p-6 pt-0", className), ...props })
);
CardFooter.displayName = "CardFooter";

// src/components/ui/button.tsx
import * as React2 from "react";
import { Slot } from "@radix-ui/react-slot";
import { cva } from "class-variance-authority";
import { jsx as jsx2 } from "react/jsx-runtime";
var buttonVariants = cva(
  "inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0",
  {
    variants: {
      variant: {
        default: "bg-primary text-primary-foreground shadow hover:bg-primary/90",
        destructive: "bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90",
        outline: "border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground",
        secondary: "bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80",
        ghost: "hover:bg-accent hover:text-accent-foreground",
        link: "text-primary underline-offset-4 hover:underline"
      },
      size: {
        default: "h-9 px-4 py-2",
        sm: "h-8 rounded-md px-3 text-xs",
        lg: "h-10 rounded-md px-8",
        icon: "h-9 w-9"
      }
    },
    defaultVariants: {
      variant: "default",
      size: "default"
    }
  }
);
var Button = React2.forwardRef(
  ({ className, variant, size, asChild = false, ...props }, ref) => {
    const Comp = asChild ? Slot : "button";
    return /* @__PURE__ */ jsx2(
      Comp,
      {
        className: cn(buttonVariants({ variant, size, className })),
        ref,
        ...props
      }
    );
  }
);
Button.displayName = "Button";

// src/components/ui/badge.tsx
import { cva as cva2 } from "class-variance-authority";
import { jsx as jsx3 } from "react/jsx-runtime";
var badgeVariants = cva2(
  "inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  {
    variants: {
      variant: {
        default: "border-transparent bg-primary text-primary-foreground shadow hover:bg-primary/80",
        secondary: "border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80",
        destructive: "border-transparent bg-destructive text-destructive-foreground shadow hover:bg-destructive/80",
        outline: "text-foreground",
        success: "border-transparent bg-emerald-500/10 text-emerald-500 hover:bg-emerald-500/20",
        warning: "border-transparent bg-amber-500/10 text-amber-500 hover:bg-amber-500/20"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
function Badge({ className, variant, ...props }) {
  return /* @__PURE__ */ jsx3("div", { className: cn(badgeVariants({ variant }), className), ...props });
}

// src/components/ai/AgentFleet.tsx
import { jsx as jsx4, jsxs } from "react/jsx-runtime";
function AgentFleet({ workers: initialWorkers, onSpawnWorker, onViewNodeGraph }) {
  const [workers, setWorkers] = useState(
    initialWorkers || [
      { id: "1", name: "MarketScraper_01", task: "Scraping tech crunch for funding news", status: "working", load: 78 },
      { id: "2", name: "LeadEnricher_01", task: "Awaiting leads from Scraper", status: "idle", load: 0 },
      { id: "3", name: "ContentDraft_01", task: "Generating social media copy", status: "working", load: 45 },
      { id: "4", name: "FinanceAuditor_01", task: "Verifying daily revenue sync", status: "complete", load: 0 }
    ]
  );
  const handleSpawn = () => {
    if (onSpawnWorker) {
      onSpawnWorker();
      return;
    }
    const newWorker = {
      id: Math.random().toString(36).slice(2, 8),
      name: `Worker_${Math.floor(Math.random() * 1e3)}`,
      task: "Initializing strategic analysis...",
      status: "idle",
      load: 0
    };
    setWorkers((prev) => [...prev, newWorker]);
  };
  const activeCount = workers.filter((w) => w.status === "working").length;
  const awaitingApproval = workers.filter((w) => w.status === "awaiting_approval").length;
  const errorCount = workers.filter((w) => w.status === "error").length;
  const avgLoad = workers.length > 0 ? Math.round(workers.reduce((sum, w) => sum + w.load, 0) / workers.length) : 0;
  return /* @__PURE__ */ jsxs("div", { className: "h-full p-8 space-y-8", children: [
    /* @__PURE__ */ jsxs("div", { className: "flex items-center justify-between border-b pb-6", children: [
      /* @__PURE__ */ jsxs("div", { children: [
        /* @__PURE__ */ jsxs("h2", { className: "text-2xl font-black text-emerald-500 tracking-tighter flex items-center gap-3 italic", children: [
          /* @__PURE__ */ jsx4(Network, { className: "h-8 w-8" }),
          "MULTI-AGENT ORCHESTRATOR"
        ] }),
        /* @__PURE__ */ jsx4("p", { className: "text-muted-foreground text-xs font-bold uppercase tracking-widest mt-1", children: "Parallel reasoning & distributed task execution" })
      ] }),
      /* @__PURE__ */ jsxs("div", { className: "flex gap-2", children: [
        /* @__PURE__ */ jsx4(Button, { variant: "outline", onClick: onViewNodeGraph, children: "View Node Graph" }),
        /* @__PURE__ */ jsxs(Button, { onClick: handleSpawn, className: "bg-emerald-500 hover:bg-emerald-600 text-black font-black uppercase text-[10px] tracking-widest", children: [
          "Spawn Worker ",
          /* @__PURE__ */ jsx4(Plus, { className: "h-3 w-3 ml-2" })
        ] })
      ] })
    ] }),
    /* @__PURE__ */ jsxs("div", { className: "grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6", children: [
      /* @__PURE__ */ jsxs(Card, { className: "border-emerald-500/30 border-2 relative overflow-hidden group", children: [
        /* @__PURE__ */ jsx4("div", { className: "absolute top-0 right-0 p-8 opacity-5 group-hover:scale-110 transition-transform", children: /* @__PURE__ */ jsx4(Cpu, { className: "h-24 w-24 text-emerald-500" }) }),
        /* @__PURE__ */ jsxs(CardHeader, { children: [
          /* @__PURE__ */ jsxs("div", { className: "flex justify-between items-start", children: [
            /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-3", children: [
              /* @__PURE__ */ jsx4("div", { className: "p-2 bg-emerald-500/10 rounded-lg", children: /* @__PURE__ */ jsx4(Zap, { className: "h-5 w-5 text-emerald-400" }) }),
              /* @__PURE__ */ jsx4(CardTitle, { className: "text-base italic uppercase tracking-widest", children: "Master Orchestrator" })
            ] }),
            /* @__PURE__ */ jsx4(Badge, { className: "bg-emerald-500 text-black animate-pulse", children: "OPTIMIZED" })
          ] }),
          /* @__PURE__ */ jsx4(CardDescription, { className: "text-muted-foreground text-[10px] uppercase font-black tracking-widest mt-2", children: "Core Coordination Logic" })
        ] }),
        /* @__PURE__ */ jsxs(CardContent, { className: "space-y-6", children: [
          /* @__PURE__ */ jsxs("div", { className: "space-y-3", children: [
            /* @__PURE__ */ jsxs("div", { className: "flex justify-between text-[10px] font-bold uppercase italic", children: [
              /* @__PURE__ */ jsx4("span", { className: "text-muted-foreground", children: "GLOBAL COORDINATION LOAD" }),
              /* @__PURE__ */ jsxs("span", { className: "text-emerald-400", children: [
                avgLoad,
                "%"
              ] })
            ] }),
            /* @__PURE__ */ jsx4("div", { className: "h-1.5 w-full bg-muted rounded-full overflow-hidden", children: /* @__PURE__ */ jsx4("div", { className: "h-full bg-emerald-500 transition-all", style: { width: `${avgLoad}%` } }) })
          ] }),
          /* @__PURE__ */ jsxs("div", { className: "grid grid-cols-2 gap-3", children: [
            /* @__PURE__ */ jsxs("div", { className: "p-3 bg-muted rounded-lg text-center", children: [
              /* @__PURE__ */ jsx4("p", { className: "text-xs font-black text-white", children: activeCount }),
              /* @__PURE__ */ jsx4("p", { className: "text-[8px] text-muted-foreground uppercase font-black", children: "ACTIVE AGENTS" })
            ] }),
            /* @__PURE__ */ jsxs("div", { className: "p-3 bg-muted rounded-lg text-center", children: [
              /* @__PURE__ */ jsx4("p", { className: "text-xs font-black text-white", children: workers.length }),
              /* @__PURE__ */ jsx4("p", { className: "text-[8px] text-muted-foreground uppercase font-black", children: "TOTAL WORKERS" })
            ] }),
            awaitingApproval > 0 && /* @__PURE__ */ jsxs("div", { className: "p-3 bg-amber-500/10 border border-amber-500/30 rounded-lg text-center", children: [
              /* @__PURE__ */ jsx4("p", { className: "text-xs font-black text-amber-500", children: awaitingApproval }),
              /* @__PURE__ */ jsx4("p", { className: "text-[8px] text-amber-500/70 uppercase font-black", children: "AWAITING APPROVAL" })
            ] }),
            errorCount > 0 && /* @__PURE__ */ jsxs("div", { className: "p-3 bg-destructive/10 border border-destructive/30 rounded-lg text-center", children: [
              /* @__PURE__ */ jsx4("p", { className: "text-xs font-black text-destructive", children: errorCount }),
              /* @__PURE__ */ jsx4("p", { className: "text-[8px] text-destructive/70 uppercase font-black", children: "ERRORS" })
            ] })
          ] })
        ] })
      ] }),
      /* @__PURE__ */ jsxs("div", { className: "col-span-1 md:col-span-2 space-y-4", children: [
        /* @__PURE__ */ jsxs("h3", { className: "text-xs font-black uppercase text-muted-foreground tracking-[0.2em] flex items-center gap-2", children: [
          /* @__PURE__ */ jsx4(Activity, { className: "h-3 w-3" }),
          " Worker Fleet Status"
        ] }),
        /* @__PURE__ */ jsx4("div", { className: "grid grid-cols-1 sm:grid-cols-2 gap-4", children: workers.map((worker) => /* @__PURE__ */ jsxs(
          "div",
          {
            className: "p-4 bg-card border rounded-xl hover:border-border transition-all flex items-center justify-between group cursor-pointer",
            children: [
              /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-4", children: [
                /* @__PURE__ */ jsxs(
                  "div",
                  {
                    className: cn(
                      "h-10 w-10 rounded-lg flex items-center justify-center transition-colors",
                      worker.status === "working" && "bg-emerald-500/10 text-emerald-500",
                      worker.status === "complete" && "bg-blue-500/10 text-blue-500",
                      worker.status === "error" && "bg-destructive/10 text-destructive",
                      worker.status === "awaiting_approval" && "bg-amber-500/10 text-amber-500",
                      worker.status === "idle" && "bg-muted text-muted-foreground"
                    ),
                    children: [
                      worker.status === "working" && /* @__PURE__ */ jsx4(Loader2, { className: "h-5 w-5 animate-spin" }),
                      worker.status === "complete" && /* @__PURE__ */ jsx4(CheckCircle2, { className: "h-5 w-5" }),
                      worker.status === "error" && /* @__PURE__ */ jsx4(AlertTriangle, { className: "h-5 w-5" }),
                      worker.status === "awaiting_approval" && /* @__PURE__ */ jsx4(Users, { className: "h-5 w-5" }),
                      worker.status === "idle" && /* @__PURE__ */ jsx4(Bot, { className: "h-5 w-5" })
                    ]
                  }
                ),
                /* @__PURE__ */ jsxs("div", { className: "space-y-0.5", children: [
                  /* @__PURE__ */ jsx4("p", { className: "text-[11px] font-black text-foreground uppercase tracking-tight", children: worker.name }),
                  /* @__PURE__ */ jsx4("p", { className: "text-[10px] text-muted-foreground line-clamp-1 italic", children: worker.task })
                ] })
              ] }),
              /* @__PURE__ */ jsxs("div", { className: "text-right", children: [
                /* @__PURE__ */ jsxs("p", { className: "text-[10px] font-black text-emerald-500", children: [
                  worker.load,
                  "%"
                ] }),
                /* @__PURE__ */ jsx4("div", { className: "h-1 w-12 bg-muted rounded-full mt-1", children: /* @__PURE__ */ jsx4("div", { className: "h-full bg-emerald-500/50 transition-all", style: { width: `${worker.load}%` } }) })
              ] })
            ]
          },
          worker.id
        )) })
      ] })
    ] }),
    /* @__PURE__ */ jsxs(Card, { children: [
      /* @__PURE__ */ jsxs("div", { className: "h-8 border-b flex items-center px-4", children: [
        /* @__PURE__ */ jsx4(TerminalIcon, { className: "h-3 w-3 text-emerald-500 mr-2" }),
        /* @__PURE__ */ jsx4("span", { className: "text-[9px] font-black text-muted-foreground uppercase tracking-[0.2em]", children: "Distributed Logic Logs" })
      ] }),
      /* @__PURE__ */ jsxs("div", { className: "p-6 font-mono text-[10px] space-y-2 text-muted-foreground leading-relaxed italic", children: [
        /* @__PURE__ */ jsxs("div", { className: "flex gap-4", children: [
          /* @__PURE__ */ jsx4("span", { className: "text-muted-foreground/30 shrink-0", children: (/* @__PURE__ */ new Date()).toLocaleTimeString() }),
          /* @__PURE__ */ jsx4("span", { children: "[MASTER] Orchestrator initialized. Fleet ready." })
        ] }),
        workers.filter((w) => w.status === "working").map((w) => /* @__PURE__ */ jsxs("div", { className: "flex gap-4", children: [
          /* @__PURE__ */ jsx4("span", { className: "text-muted-foreground/30 shrink-0", children: (/* @__PURE__ */ new Date()).toLocaleTimeString() }),
          /* @__PURE__ */ jsxs("span", { className: "text-emerald-500", children: [
            "[",
            w.name,
            "] ",
            w.task
          ] })
        ] }, w.id)),
        workers.filter((w) => w.status === "awaiting_approval").map((w) => /* @__PURE__ */ jsxs("div", { className: "flex gap-4", children: [
          /* @__PURE__ */ jsx4("span", { className: "text-muted-foreground/30 shrink-0", children: (/* @__PURE__ */ new Date()).toLocaleTimeString() }),
          /* @__PURE__ */ jsxs("span", { className: "text-amber-500", children: [
            "[",
            w.name,
            "] Awaiting human approval..."
          ] })
        ] }, w.id))
      ] })
    ] })
  ] });
}
export {
  AgentFleet
};
