// @billion-business/venture-ui/dist/components/ai/BrainInterface.js
'use client';

// src/components/ai/BrainInterface.tsx
import { useState, useEffect, useRef } from "react";
import { Brain, Terminal, Cpu, Send, Loader2, ShieldAlert, CheckCircle2, Info, ChevronRight } from "lucide-react";

// src/components/ui/card.tsx
import * as React from "react";

// src/lib/utils.ts
import { clsx } from "clsx";
import { twMerge } from "tailwind-merge";
function cn(...inputs) {
  return twMerge(clsx(inputs));
}

// src/components/ui/card.tsx
import { jsx } from "react/jsx-runtime";
var Card = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      className: cn("rounded-xl border bg-card text-card-foreground shadow", className),
      ...props
    }
  )
);
Card.displayName = "Card";
var CardHeader = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("flex flex-col space-y-1.5 p-6", className), ...props })
);
CardHeader.displayName = "CardHeader";
var CardTitle = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("font-semibold leading-none tracking-tight", className), ...props })
);
CardTitle.displayName = "CardTitle";
var CardDescription = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("text-sm text-muted-foreground", className), ...props })
);
CardDescription.displayName = "CardDescription";
var CardContent = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("p-6 pt-0", className), ...props })
);
CardContent.displayName = "CardContent";
var CardFooter = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("flex items-center p-6 pt-0", className), ...props })
);
CardFooter.displayName = "CardFooter";

// src/components/ui/button.tsx
import * as React2 from "react";
import { Slot } from "@radix-ui/react-slot";
import { cva } from "class-variance-authority";
import { jsx as jsx2 } from "react/jsx-runtime";
var buttonVariants = cva(
  "inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0",
  {
    variants: {
      variant: {
        default: "bg-primary text-primary-foreground shadow hover:bg-primary/90",
        destructive: "bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90",
        outline: "border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground",
        secondary: "bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80",
        ghost: "hover:bg-accent hover:text-accent-foreground",
        link: "text-primary underline-offset-4 hover:underline"
      },
      size: {
        default: "h-9 px-4 py-2",
        sm: "h-8 rounded-md px-3 text-xs",
        lg: "h-10 rounded-md px-8",
        icon: "h-9 w-9"
      }
    },
    defaultVariants: {
      variant: "default",
      size: "default"
    }
  }
);
var Button = React2.forwardRef(
  ({ className, variant, size, asChild = false, ...props }, ref) => {
    const Comp = asChild ? Slot : "button";
    return /* @__PURE__ */ jsx2(
      Comp,
      {
        className: cn(buttonVariants({ variant, size, className })),
        ref,
        ...props
      }
    );
  }
);
Button.displayName = "Button";

// src/components/ui/input.tsx
import * as React3 from "react";
import { jsx as jsx3 } from "react/jsx-runtime";
var Input = React3.forwardRef(
  ({ className, type, ...props }, ref) => {
    return /* @__PURE__ */ jsx3(
      "input",
      {
        type,
        className: cn(
          "flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm",
          className
        ),
        ref,
        ...props
      }
    );
  }
);
Input.displayName = "Input";

// src/components/ui/scroll-area.tsx
import * as React4 from "react";
import * as ScrollAreaPrimitive from "@radix-ui/react-scroll-area";
import { jsx as jsx4, jsxs } from "react/jsx-runtime";
var ScrollArea = React4.forwardRef(({ className, children, ...props }, ref) => /* @__PURE__ */ jsxs(
  ScrollAreaPrimitive.Root,
  {
    ref,
    className: cn("relative overflow-hidden", className),
    ...props,
    children: [
      /* @__PURE__ */ jsx4(ScrollAreaPrimitive.Viewport, { className: "h-full w-full rounded-[inherit]", children }),
      /* @__PURE__ */ jsx4(ScrollBar, {}),
      /* @__PURE__ */ jsx4(ScrollAreaPrimitive.Corner, {})
    ]
  }
));
ScrollArea.displayName = ScrollAreaPrimitive.Root.displayName;
var ScrollBar = React4.forwardRef(({ className, orientation = "vertical", ...props }, ref) => /* @__PURE__ */ jsx4(
  ScrollAreaPrimitive.ScrollAreaScrollbar,
  {
    ref,
    orientation,
    className: cn(
      "flex touch-none select-none transition-colors",
      orientation === "vertical" && "h-full w-2.5 border-l border-l-transparent p-[1px]",
      orientation === "horizontal" && "h-2.5 flex-col border-t border-t-transparent p-[1px]",
      className
    ),
    ...props,
    children: /* @__PURE__ */ jsx4(ScrollAreaPrimitive.ScrollAreaThumb, { className: "relative flex-1 rounded-full bg-border" })
  }
));
ScrollBar.displayName = ScrollAreaPrimitive.ScrollAreaScrollbar.displayName;

// src/components/ui/badge.tsx
import { cva as cva2 } from "class-variance-authority";
import { jsx as jsx5 } from "react/jsx-runtime";
var badgeVariants = cva2(
  "inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  {
    variants: {
      variant: {
        default: "border-transparent bg-primary text-primary-foreground shadow hover:bg-primary/80",
        secondary: "border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80",
        destructive: "border-transparent bg-destructive text-destructive-foreground shadow hover:bg-destructive/80",
        outline: "text-foreground",
        success: "border-transparent bg-emerald-500/10 text-emerald-500 hover:bg-emerald-500/20",
        warning: "border-transparent bg-amber-500/10 text-amber-500 hover:bg-amber-500/20"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
function Badge({ className, variant, ...props }) {
  return /* @__PURE__ */ jsx5("div", { className: cn(badgeVariants({ variant }), className), ...props });
}

// src/components/ai/BrainInterface.tsx
import { motion } from "motion/react";
import { jsx as jsx6, jsxs as jsxs2 } from "react/jsx-runtime";
function BrainInterface({ onSend, initialMessage, mode = "general" }) {
  const [messages, setMessages] = useState([
    {
      role: "agent",
      content: initialMessage || "Core reasoning engine initialized. How can I assist?",
      timestamp: (/* @__PURE__ */ new Date()).toLocaleTimeString(),
      type: "system"
    }
  ]);
  const [input, setInput] = useState("");
  const [isReasoning, setIsReasoning] = useState(false);
  const [reasoningLog, setReasoningLog] = useState(null);
  const scrollRef = useRef(null);
  useEffect(() => {
    if (scrollRef.current) {
      scrollRef.current.scrollTop = scrollRef.current.scrollHeight;
    }
  }, [messages]);
  const handleSend = async () => {
    if (!input.trim()) return;
    const userMessage = {
      role: "user",
      content: input,
      timestamp: (/* @__PURE__ */ new Date()).toLocaleTimeString()
    };
    setMessages((prev) => [...prev, userMessage]);
    setInput("");
    setIsReasoning(true);
    try {
      const result = onSend ? await onSend(input) : {
        summary: "Local mode \u2014 connect an AI provider for full reasoning.",
        threats: [],
        recommendations: ["Configure AI provider in environment"],
        threatLevel: 0,
        reasoning: "No AI provider connected."
      };
      setReasoningLog(result);
      const agentMessage = {
        role: "agent",
        content: result.summary,
        timestamp: (/* @__PURE__ */ new Date()).toLocaleTimeString(),
        recommendations: result.recommendations,
        threatLevel: result.threatLevel
      };
      setMessages((prev) => [...prev, agentMessage]);
    } catch (error) {
      setMessages((prev) => [
        ...prev,
        {
          role: "agent",
          content: "Error in reasoning engine.",
          timestamp: (/* @__PURE__ */ new Date()).toLocaleTimeString(),
          type: "system"
        }
      ]);
    } finally {
      setIsReasoning(false);
    }
  };
  const modeConfig = {
    security: { title: "AI Security Agent", subtitle: "LLM-BASED REASONING ENGINE v2.0", badge: "Autonomous Mode", badgeColor: "text-emerald-500" },
    general: { title: "AI Assistant", subtitle: "REASONING ENGINE", badge: "Active", badgeColor: "text-blue-500" },
    oversight: { title: "Oversight Agent", subtitle: "HUMAN-IN-THE-LOOP MONITOR", badge: "Supervising", badgeColor: "text-amber-500" }
  };
  const config = modeConfig[mode];
  return /* @__PURE__ */ jsxs2("div", { className: "grid grid-cols-1 lg:grid-cols-3 gap-6 h-[calc(100vh-12rem)]", children: [
    /* @__PURE__ */ jsxs2(Card, { className: "lg:col-span-2 flex flex-col", children: [
      /* @__PURE__ */ jsx6(CardHeader, { className: "border-b", children: /* @__PURE__ */ jsxs2("div", { className: "flex items-center justify-between", children: [
        /* @__PURE__ */ jsxs2("div", { className: "flex items-center gap-3", children: [
          /* @__PURE__ */ jsx6("div", { className: "p-2 rounded-lg bg-primary/10", children: /* @__PURE__ */ jsx6(Brain, { className: "w-5 h-5 text-primary" }) }),
          /* @__PURE__ */ jsxs2("div", { children: [
            /* @__PURE__ */ jsx6(CardTitle, { className: "text-lg", children: config.title }),
            /* @__PURE__ */ jsx6(CardDescription, { className: "text-xs font-mono", children: config.subtitle })
          ] })
        ] }),
        /* @__PURE__ */ jsx6(Badge, { variant: "outline", className: `font-mono text-[10px] uppercase tracking-widest border-emerald-500/30 ${config.badgeColor}`, children: config.badge })
      ] }) }),
      /* @__PURE__ */ jsx6(CardContent, { className: "flex-1 p-0 overflow-hidden", children: /* @__PURE__ */ jsx6(ScrollArea, { className: "h-full p-6", ref: scrollRef, children: /* @__PURE__ */ jsxs2("div", { className: "space-y-6", children: [
        messages.map((msg, i) => /* @__PURE__ */ jsx6(
          motion.div,
          {
            initial: { opacity: 0, x: msg.role === "user" ? 10 : -10 },
            animate: { opacity: 1, x: 0 },
            className: `flex ${msg.role === "user" ? "justify-end" : "justify-start"}`,
            children: /* @__PURE__ */ jsxs2("div", { className: `max-w-[80%] space-y-2 ${msg.role === "user" ? "items-end" : "items-start"}`, children: [
              /* @__PURE__ */ jsxs2(
                "div",
                {
                  className: `p-4 rounded-2xl text-sm ${msg.role === "user" ? "bg-primary text-primary-foreground rounded-tr-none" : "bg-muted/50 border rounded-tl-none"}`,
                  children: [
                    msg.content,
                    msg.recommendations && /* @__PURE__ */ jsxs2("div", { className: "mt-4 space-y-2", children: [
                      /* @__PURE__ */ jsx6("p", { className: "text-[10px] font-bold uppercase tracking-widest opacity-70", children: "Recommended Actions:" }),
                      msg.recommendations.map((rec, j) => /* @__PURE__ */ jsxs2("div", { className: "flex items-start gap-2 text-xs bg-background/30 p-2 rounded-lg border border-white/5", children: [
                        /* @__PURE__ */ jsx6(ChevronRight, { className: "w-3 h-3 mt-0.5 text-primary shrink-0" }),
                        /* @__PURE__ */ jsx6("span", { children: rec })
                      ] }, j))
                    ] })
                  ]
                }
              ),
              /* @__PURE__ */ jsx6("p", { className: "text-[10px] text-muted-foreground font-mono px-1", children: msg.timestamp })
            ] })
          },
          i
        )),
        isReasoning && /* @__PURE__ */ jsx6("div", { className: "flex justify-start", children: /* @__PURE__ */ jsxs2("div", { className: "bg-muted/30 border p-4 rounded-2xl rounded-tl-none flex items-center gap-3", children: [
          /* @__PURE__ */ jsx6(Loader2, { className: "w-4 h-4 animate-spin text-primary" }),
          /* @__PURE__ */ jsx6("span", { className: "text-xs font-mono animate-pulse", children: "Agent is reasoning..." })
        ] }) })
      ] }) }) }),
      /* @__PURE__ */ jsx6(CardFooter, { className: "p-4 border-t", children: /* @__PURE__ */ jsxs2("div", { className: "flex w-full gap-3", children: [
        /* @__PURE__ */ jsx6(
          Input,
          {
            placeholder: "Ask the agent...",
            value: input,
            onChange: (e) => setInput(e.target.value),
            onKeyDown: (e) => e.key === "Enter" && handleSend(),
            className: "bg-background/50"
          }
        ),
        /* @__PURE__ */ jsx6(Button, { onClick: handleSend, disabled: isReasoning || !input.trim(), children: /* @__PURE__ */ jsx6(Send, { className: "w-4 h-4" }) })
      ] }) })
    ] }),
    /* @__PURE__ */ jsxs2("div", { className: "space-y-6", children: [
      /* @__PURE__ */ jsxs2(Card, { children: [
        /* @__PURE__ */ jsx6(CardHeader, { className: "pb-2", children: /* @__PURE__ */ jsxs2("div", { className: "flex items-center gap-2", children: [
          /* @__PURE__ */ jsx6(Terminal, { className: "w-4 h-4 text-primary" }),
          /* @__PURE__ */ jsx6(CardTitle, { className: "text-sm font-mono uppercase tracking-widest", children: "Reasoning Log" })
        ] }) }),
        /* @__PURE__ */ jsx6(CardContent, { className: "space-y-4", children: reasoningLog ? /* @__PURE__ */ jsxs2("div", { className: "space-y-4", children: [
          /* @__PURE__ */ jsxs2("div", { className: "p-3 rounded-lg bg-background/50 border font-mono text-[11px] leading-relaxed", children: [
            /* @__PURE__ */ jsx6("span", { className: "text-primary font-bold", children: "LOG_ENTRY:" }),
            " ",
            reasoningLog.reasoning
          ] }),
          /* @__PURE__ */ jsxs2("div", { className: "grid grid-cols-2 gap-3", children: [
            /* @__PURE__ */ jsxs2("div", { className: "p-3 rounded-lg bg-background/50 border text-center", children: [
              /* @__PURE__ */ jsx6("p", { className: "text-[10px] text-muted-foreground uppercase font-mono mb-1", children: "Threat Level" }),
              /* @__PURE__ */ jsx6("p", { className: `text-xl font-bold ${reasoningLog.threatLevel > 50 ? "text-destructive" : "text-emerald-500"}`, children: reasoningLog.threatLevel })
            ] }),
            /* @__PURE__ */ jsxs2("div", { className: "p-3 rounded-lg bg-background/50 border text-center", children: [
              /* @__PURE__ */ jsx6("p", { className: "text-[10px] text-muted-foreground uppercase font-mono mb-1", children: "Confidence" }),
              /* @__PURE__ */ jsx6("p", { className: "text-xl font-bold text-primary", children: "94%" })
            ] })
          ] })
        ] }) : /* @__PURE__ */ jsxs2("div", { className: "h-32 flex flex-col items-center justify-center text-muted-foreground opacity-50 border-2 border-dashed rounded-xl", children: [
          /* @__PURE__ */ jsx6(Cpu, { className: "w-8 h-8 mb-2" }),
          /* @__PURE__ */ jsx6("p", { className: "text-xs font-mono", children: "Waiting for inference..." })
        ] }) })
      ] }),
      /* @__PURE__ */ jsxs2(Card, { children: [
        /* @__PURE__ */ jsx6(CardHeader, { className: "pb-2", children: /* @__PURE__ */ jsx6(CardTitle, { className: "text-sm font-semibold", children: "Active Guardrails" }) }),
        /* @__PURE__ */ jsxs2(CardContent, { className: "space-y-3", children: [
          /* @__PURE__ */ jsx6(GuardrailItem, { icon: CheckCircle2, label: "Explainable Decisions", status: "Active", color: "text-emerald-500" }),
          /* @__PURE__ */ jsx6(GuardrailItem, { icon: ShieldAlert, label: "Human-in-the-loop", status: "Threshold: High", color: "text-amber-500" }),
          /* @__PURE__ */ jsx6(GuardrailItem, { icon: Info, label: "Immutable Audit Trail", status: "Encrypted", color: "text-blue-500" })
        ] })
      ] })
    ] })
  ] });
}
function GuardrailItem({ icon: Icon, label, status, color }) {
  return /* @__PURE__ */ jsxs2("div", { className: "flex items-center justify-between p-2 rounded-lg bg-muted/30 border border-border/20", children: [
    /* @__PURE__ */ jsxs2("div", { className: "flex items-center gap-2", children: [
      /* @__PURE__ */ jsx6(Icon, { className: `w-3.5 h-3.5 ${color}` }),
      /* @__PURE__ */ jsx6("span", { className: "text-xs font-medium", children: label })
    ] }),
    /* @__PURE__ */ jsx6("span", { className: "text-[10px] font-mono text-muted-foreground", children: status })
  ] });
}
export {
  BrainInterface
};
