// @billion-business/ai-core/src/types.ts
export type ProviderType = "gemini" | "openai" | "anthropic" | "ollama" | "deepseek"

export type TaskType =
  | "coding"
  | "business_logic"
  | "agent_brain"
  | "cross_domain"
  | "creative"
  | "analysis"
  | "summarization"
  | "chat"

export type OversightMode = "shadow" | "checkpoint" | "recovery"

export interface Message {
  role: "system" | "user" | "assistant"
  content: string
}

export interface GenerateOptions {
  model?: string
  temperature?: number
  maxTokens?: number
  responseFormat?: "text" | "json"
  systemPrompt?: string
  timeout?: number
  metadata?: Record<string, any>
}

export interface GenerateResult {
  text: string
  provider: ProviderType
  model: string
  tokensUsed?: number
  costUsd?: number
  latencyMs: number
  confidence?: number
  flagged?: boolean
  flagReason?: string
}

export interface ModelRoute {
  taskType: TaskType
  provider: ProviderType
  model: string
  fallback?: { provider: ProviderType; model: string }
  maxCostPerCall?: number
}

export interface AICoreConfig {
  providers: {
    gemini?: { apiKey: string }
    openai?: { apiKey: string; baseUrl?: string }
    anthropic?: { apiKey: string }
    ollama?: { baseUrl?: string; model?: string }
    deepseek?: { apiKey: string }
  }
  routes?: ModelRoute[]
  oversight?: {
    mode: OversightMode
    confidenceThreshold: number
    onEscalate?: (payload: EscalationPayload) => Promise<boolean>
  }
  rateLimit?: {
    maxCallsPerMinute: number
    maxTokensPerMinute: number
  }
  costTracking?: {
    enabled: boolean
    budgetLimitUsd?: number
  }
}

export interface EscalationPayload {
  action: "generate" | "chat" | "analyze"
  provider: ProviderType
  model: string
  prompt: string
  confidence: number
  risk: "low" | "medium" | "high" | "critical"
  metadata?: Record<string, any>
}

export interface UsageStats {
  totalCalls: number
  totalTokens: number
  totalCostUsd: number
  callsByProvider: Record<ProviderType, number>
  callsByTask: Record<TaskType, number>
  averageLatencyMs: number
  errorCount: number
}
