// @billion-business/venture-ui/dist/index.js
'use client';

// src/lib/utils.ts
import { clsx } from "clsx";
import { twMerge } from "tailwind-merge";
function cn(...inputs) {
  return twMerge(clsx(inputs));
}

// src/components/ui/button.tsx
import * as React from "react";
import { Slot } from "@radix-ui/react-slot";
import { cva } from "class-variance-authority";
import { jsx } from "react/jsx-runtime";
var buttonVariants = cva(
  "inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0",
  {
    variants: {
      variant: {
        default: "bg-primary text-primary-foreground shadow hover:bg-primary/90",
        destructive: "bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90",
        outline: "border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground",
        secondary: "bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80",
        ghost: "hover:bg-accent hover:text-accent-foreground",
        link: "text-primary underline-offset-4 hover:underline"
      },
      size: {
        default: "h-9 px-4 py-2",
        sm: "h-8 rounded-md px-3 text-xs",
        lg: "h-10 rounded-md px-8",
        icon: "h-9 w-9"
      }
    },
    defaultVariants: {
      variant: "default",
      size: "default"
    }
  }
);
var Button = React.forwardRef(
  ({ className, variant, size, asChild = false, ...props }, ref) => {
    const Comp = asChild ? Slot : "button";
    return /* @__PURE__ */ jsx(
      Comp,
      {
        className: cn(buttonVariants({ variant, size, className })),
        ref,
        ...props
      }
    );
  }
);
Button.displayName = "Button";

// src/components/ui/card.tsx
import * as React2 from "react";
import { jsx as jsx2 } from "react/jsx-runtime";
var Card = React2.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx2(
    "div",
    {
      ref,
      className: cn("rounded-xl border bg-card text-card-foreground shadow", className),
      ...props
    }
  )
);
Card.displayName = "Card";
var CardHeader = React2.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx2("div", { ref, className: cn("flex flex-col space-y-1.5 p-6", className), ...props })
);
CardHeader.displayName = "CardHeader";
var CardTitle = React2.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx2("div", { ref, className: cn("font-semibold leading-none tracking-tight", className), ...props })
);
CardTitle.displayName = "CardTitle";
var CardDescription = React2.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx2("div", { ref, className: cn("text-sm text-muted-foreground", className), ...props })
);
CardDescription.displayName = "CardDescription";
var CardContent = React2.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx2("div", { ref, className: cn("p-6 pt-0", className), ...props })
);
CardContent.displayName = "CardContent";
var CardFooter = React2.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx2("div", { ref, className: cn("flex items-center p-6 pt-0", className), ...props })
);
CardFooter.displayName = "CardFooter";

// src/components/ui/input.tsx
import * as React3 from "react";
import { jsx as jsx3 } from "react/jsx-runtime";
var Input = React3.forwardRef(
  ({ className, type, ...props }, ref) => {
    return /* @__PURE__ */ jsx3(
      "input",
      {
        type,
        className: cn(
          "flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm",
          className
        ),
        ref,
        ...props
      }
    );
  }
);
Input.displayName = "Input";

// src/components/ui/badge.tsx
import { cva as cva2 } from "class-variance-authority";
import { jsx as jsx4 } from "react/jsx-runtime";
var badgeVariants = cva2(
  "inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  {
    variants: {
      variant: {
        default: "border-transparent bg-primary text-primary-foreground shadow hover:bg-primary/80",
        secondary: "border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80",
        destructive: "border-transparent bg-destructive text-destructive-foreground shadow hover:bg-destructive/80",
        outline: "text-foreground",
        success: "border-transparent bg-emerald-500/10 text-emerald-500 hover:bg-emerald-500/20",
        warning: "border-transparent bg-amber-500/10 text-amber-500 hover:bg-amber-500/20"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
function Badge({ className, variant, ...props }) {
  return /* @__PURE__ */ jsx4("div", { className: cn(badgeVariants({ variant }), className), ...props });
}

// src/components/ui/dialog.tsx
import * as DialogPrimitive from "@radix-ui/react-dialog";
import { X } from "lucide-react";
import { jsx as jsx5, jsxs } from "react/jsx-runtime";
var Dialog = DialogPrimitive.Root;
var DialogTrigger = DialogPrimitive.Trigger;
var DialogClose = DialogPrimitive.Close;
function DialogPortal({ children, ...props }) {
  return /* @__PURE__ */ jsx5(DialogPrimitive.Portal, { ...props, children });
}
DialogPortal.displayName = DialogPrimitive.Portal.displayName;
function DialogOverlay({ className, ...props }) {
  return /* @__PURE__ */ jsx5(
    DialogPrimitive.Overlay,
    {
      className: cn(
        "fixed inset-0 z-50 bg-black/80 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0",
        className
      ),
      ...props
    }
  );
}
DialogOverlay.displayName = DialogPrimitive.Overlay.displayName;
function DialogContent({ className, children, ...props }) {
  return /* @__PURE__ */ jsxs(DialogPortal, { children: [
    /* @__PURE__ */ jsx5(DialogOverlay, {}),
    /* @__PURE__ */ jsxs(
      DialogPrimitive.Content,
      {
        className: cn(
          "fixed left-[50%] top-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border bg-background p-6 shadow-lg duration-200 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] sm:rounded-lg",
          className
        ),
        ...props,
        children: [
          children,
          /* @__PURE__ */ jsxs(DialogPrimitive.Close, { className: "absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground", children: [
            /* @__PURE__ */ jsx5(X, { className: "h-4 w-4" }),
            /* @__PURE__ */ jsx5("span", { className: "sr-only", children: "Close" })
          ] })
        ]
      }
    )
  ] });
}
DialogContent.displayName = DialogPrimitive.Content.displayName;
function DialogHeader({ className, ...props }) {
  return /* @__PURE__ */ jsx5("div", { className: cn("flex flex-col space-y-1.5 text-center sm:text-left", className), ...props });
}
DialogHeader.displayName = "DialogHeader";
function DialogFooter({ className, ...props }) {
  return /* @__PURE__ */ jsx5("div", { className: cn("flex flex-col-reverse sm:flex-row sm:justify-end sm:space-x-2", className), ...props });
}
DialogFooter.displayName = "DialogFooter";
function DialogTitle({ className, ...props }) {
  return /* @__PURE__ */ jsx5(DialogPrimitive.Title, { className: cn("text-lg font-semibold leading-none tracking-tight", className), ...props });
}
DialogTitle.displayName = DialogPrimitive.Title.displayName;
function DialogDescription({ className, ...props }) {
  return /* @__PURE__ */ jsx5(DialogPrimitive.Description, { className: cn("text-sm text-muted-foreground", className), ...props });
}
DialogDescription.displayName = DialogPrimitive.Description.displayName;

// src/components/ui/tabs.tsx
import * as TabsPrimitive from "@radix-ui/react-tabs";
import { jsx as jsx6 } from "react/jsx-runtime";
var Tabs = TabsPrimitive.Root;
function TabsList({ className, ...props }) {
  return /* @__PURE__ */ jsx6(
    TabsPrimitive.List,
    {
      className: cn(
        "inline-flex h-9 items-center justify-center rounded-lg bg-muted p-1 text-muted-foreground",
        className
      ),
      ...props
    }
  );
}
TabsList.displayName = TabsPrimitive.List.displayName;
function TabsTrigger({ className, ...props }) {
  return /* @__PURE__ */ jsx6(
    TabsPrimitive.Trigger,
    {
      className: cn(
        "inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow",
        className
      ),
      ...props
    }
  );
}
TabsTrigger.displayName = TabsPrimitive.Trigger.displayName;
function TabsContent({ className, ...props }) {
  return /* @__PURE__ */ jsx6(
    TabsPrimitive.Content,
    {
      className: cn(
        "mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2",
        className
      ),
      ...props
    }
  );
}
TabsContent.displayName = TabsPrimitive.Content.displayName;

// src/components/ui/switch.tsx
import * as React4 from "react";
import * as SwitchPrimitives from "@radix-ui/react-switch";
import { jsx as jsx7 } from "react/jsx-runtime";
var Switch = React4.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx7(
  SwitchPrimitives.Root,
  {
    className: cn(
      "peer inline-flex h-5 w-9 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent shadow-sm transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input",
      className
    ),
    ...props,
    ref,
    children: /* @__PURE__ */ jsx7(
      SwitchPrimitives.Thumb,
      {
        className: cn(
          "pointer-events-none block h-4 w-4 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-4 data-[state=unchecked]:translate-x-0"
        )
      }
    )
  }
));
Switch.displayName = SwitchPrimitives.Root.displayName;

// src/components/ui/select.tsx
import * as SelectPrimitive from "@radix-ui/react-select";
import { Check, ChevronDown, ChevronUp } from "lucide-react";
import { jsx as jsx8, jsxs as jsxs2 } from "react/jsx-runtime";
var Select = SelectPrimitive.Root;
var SelectGroup = SelectPrimitive.Group;
var SelectValue = SelectPrimitive.Value;
function SelectTrigger({ className, children, ...props }) {
  return /* @__PURE__ */ jsxs2(
    SelectPrimitive.Trigger,
    {
      className: cn(
        "flex h-9 w-full items-center justify-between whitespace-nowrap rounded-md border border-input bg-transparent px-3 py-2 text-sm shadow-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1",
        className
      ),
      ...props,
      children: [
        children,
        /* @__PURE__ */ jsx8(SelectPrimitive.Icon, { asChild: true, children: /* @__PURE__ */ jsx8(ChevronDown, { className: "h-4 w-4 opacity-50" }) })
      ]
    }
  );
}
SelectTrigger.displayName = SelectPrimitive.Trigger.displayName;
function SelectScrollUpButton({ className, ...props }) {
  return /* @__PURE__ */ jsx8(
    SelectPrimitive.ScrollUpButton,
    {
      className: cn("flex cursor-default items-center justify-center py-1", className),
      ...props,
      children: /* @__PURE__ */ jsx8(ChevronUp, { className: "h-4 w-4" })
    }
  );
}
SelectScrollUpButton.displayName = SelectPrimitive.ScrollUpButton.displayName;
function SelectScrollDownButton({ className, ...props }) {
  return /* @__PURE__ */ jsx8(
    SelectPrimitive.ScrollDownButton,
    {
      className: cn("flex cursor-default items-center justify-center py-1", className),
      ...props,
      children: /* @__PURE__ */ jsx8(ChevronDown, { className: "h-4 w-4" })
    }
  );
}
SelectScrollDownButton.displayName = SelectPrimitive.ScrollDownButton.displayName;
function SelectContent({ className, children, position = "popper", ...props }) {
  return /* @__PURE__ */ jsx8(SelectPrimitive.Portal, { children: /* @__PURE__ */ jsxs2(
    SelectPrimitive.Content,
    {
      className: cn(
        "relative z-50 max-h-96 min-w-[8rem] overflow-hidden rounded-md border bg-popover text-popover-foreground shadow-md data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2",
        position === "popper" && "data-[side=bottom]:translate-y-1 data-[side=left]:-translate-x-1 data-[side=right]:translate-x-1 data-[side=top]:-translate-y-1",
        className
      ),
      position,
      ...props,
      children: [
        /* @__PURE__ */ jsx8(SelectScrollUpButton, {}),
        /* @__PURE__ */ jsx8(
          SelectPrimitive.Viewport,
          {
            className: cn(
              "p-1",
              position === "popper" && "h-[var(--radix-select-trigger-height)] w-full min-w-[var(--radix-select-trigger-width)]"
            ),
            children
          }
        ),
        /* @__PURE__ */ jsx8(SelectScrollDownButton, {})
      ]
    }
  ) });
}
SelectContent.displayName = SelectPrimitive.Content.displayName;
function SelectLabel({ className, ...props }) {
  return /* @__PURE__ */ jsx8(SelectPrimitive.Label, { className: cn("px-2 py-1.5 text-sm font-semibold", className), ...props });
}
SelectLabel.displayName = SelectPrimitive.Label.displayName;
function SelectItem({ className, children, ...props }) {
  return /* @__PURE__ */ jsxs2(
    SelectPrimitive.Item,
    {
      className: cn(
        "relative flex w-full cursor-default select-none items-center rounded-sm py-1.5 pl-2 pr-8 text-sm outline-none focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50",
        className
      ),
      ...props,
      children: [
        /* @__PURE__ */ jsx8("span", { className: "absolute right-2 flex h-3.5 w-3.5 items-center justify-center", children: /* @__PURE__ */ jsx8(SelectPrimitive.ItemIndicator, { children: /* @__PURE__ */ jsx8(Check, { className: "h-4 w-4" }) }) }),
        /* @__PURE__ */ jsx8(SelectPrimitive.ItemText, { children })
      ]
    }
  );
}
SelectItem.displayName = SelectPrimitive.Item.displayName;
function SelectSeparator({ className, ...props }) {
  return /* @__PURE__ */ jsx8(SelectPrimitive.Separator, { className: cn("-mx-1 my-1 h-px bg-muted", className), ...props });
}
SelectSeparator.displayName = SelectPrimitive.Separator.displayName;

// src/components/ui/scroll-area.tsx
import * as React5 from "react";
import * as ScrollAreaPrimitive from "@radix-ui/react-scroll-area";
import { jsx as jsx9, jsxs as jsxs3 } from "react/jsx-runtime";
var ScrollArea = React5.forwardRef(({ className, children, ...props }, ref) => /* @__PURE__ */ jsxs3(
  ScrollAreaPrimitive.Root,
  {
    ref,
    className: cn("relative overflow-hidden", className),
    ...props,
    children: [
      /* @__PURE__ */ jsx9(ScrollAreaPrimitive.Viewport, { className: "h-full w-full rounded-[inherit]", children }),
      /* @__PURE__ */ jsx9(ScrollBar, {}),
      /* @__PURE__ */ jsx9(ScrollAreaPrimitive.Corner, {})
    ]
  }
));
ScrollArea.displayName = ScrollAreaPrimitive.Root.displayName;
var ScrollBar = React5.forwardRef(({ className, orientation = "vertical", ...props }, ref) => /* @__PURE__ */ jsx9(
  ScrollAreaPrimitive.ScrollAreaScrollbar,
  {
    ref,
    orientation,
    className: cn(
      "flex touch-none select-none transition-colors",
      orientation === "vertical" && "h-full w-2.5 border-l border-l-transparent p-[1px]",
      orientation === "horizontal" && "h-2.5 flex-col border-t border-t-transparent p-[1px]",
      className
    ),
    ...props,
    children: /* @__PURE__ */ jsx9(ScrollAreaPrimitive.ScrollAreaThumb, { className: "relative flex-1 rounded-full bg-border" })
  }
));
ScrollBar.displayName = ScrollAreaPrimitive.ScrollAreaScrollbar.displayName;

// src/components/ui/separator.tsx
import * as React6 from "react";
import * as SeparatorPrimitive from "@radix-ui/react-separator";
import { jsx as jsx10 } from "react/jsx-runtime";
var Separator2 = React6.forwardRef(({ className, orientation = "horizontal", decorative = true, ...props }, ref) => /* @__PURE__ */ jsx10(
  SeparatorPrimitive.Root,
  {
    ref,
    decorative,
    orientation,
    className: cn(
      "shrink-0 bg-border",
      orientation === "horizontal" ? "h-[1px] w-full" : "h-full w-[1px]",
      className
    ),
    ...props
  }
));
Separator2.displayName = SeparatorPrimitive.Root.displayName;

// src/components/ui/progress.tsx
import * as React7 from "react";
import * as ProgressPrimitive from "@radix-ui/react-progress";
import { jsx as jsx11 } from "react/jsx-runtime";
var Progress = React7.forwardRef(({ className, value, ...props }, ref) => /* @__PURE__ */ jsx11(
  ProgressPrimitive.Root,
  {
    ref,
    className: cn("relative h-2 w-full overflow-hidden rounded-full bg-primary/20", className),
    ...props,
    children: /* @__PURE__ */ jsx11(
      ProgressPrimitive.Indicator,
      {
        className: "h-full w-full flex-1 bg-primary transition-all",
        style: { transform: `translateX(-${100 - (value || 0)}%)` }
      }
    )
  }
));
Progress.displayName = ProgressPrimitive.Root.displayName;

// src/components/ui/textarea.tsx
import * as React8 from "react";
import { jsx as jsx12 } from "react/jsx-runtime";
var Textarea = React8.forwardRef(
  ({ className, ...props }, ref) => {
    return /* @__PURE__ */ jsx12(
      "textarea",
      {
        className: cn(
          "flex min-h-[60px] w-full rounded-md border border-input bg-transparent px-3 py-2 text-base shadow-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm",
          className
        ),
        ref,
        ...props
      }
    );
  }
);
Textarea.displayName = "Textarea";

// src/components/ui/label.tsx
import * as React9 from "react";
import { jsx as jsx13 } from "react/jsx-runtime";
var Label2 = React9.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx13(
    "label",
    {
      ref,
      className: cn(
        "text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70",
        className
      ),
      ...props
    }
  )
);
Label2.displayName = "Label";

// src/components/ui/avatar.tsx
import * as React10 from "react";
import * as AvatarPrimitive from "@radix-ui/react-avatar";
import { jsx as jsx14 } from "react/jsx-runtime";
var Avatar = React10.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx14(
  AvatarPrimitive.Root,
  {
    ref,
    className: cn("relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full", className),
    ...props
  }
));
Avatar.displayName = AvatarPrimitive.Root.displayName;
var AvatarImage = React10.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx14(AvatarPrimitive.Image, { ref, className: cn("aspect-square h-full w-full", className), ...props }));
AvatarImage.displayName = AvatarPrimitive.Image.displayName;
var AvatarFallback = React10.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx14(
  AvatarPrimitive.Fallback,
  {
    ref,
    className: cn("flex h-full w-full items-center justify-center rounded-full bg-muted", className),
    ...props
  }
));
AvatarFallback.displayName = AvatarPrimitive.Fallback.displayName;

// src/components/ai/BrainInterface.tsx
import { useState, useEffect, useRef } from "react";
import { Brain, Terminal, Cpu, Send, Loader2, ShieldAlert, CheckCircle2, Info, ChevronRight } from "lucide-react";
import { motion } from "motion/react";
import { jsx as jsx15, jsxs as jsxs4 } from "react/jsx-runtime";
function BrainInterface({ onSend, initialMessage, mode = "general" }) {
  const [messages, setMessages] = useState([
    {
      role: "agent",
      content: initialMessage || "Core reasoning engine initialized. How can I assist?",
      timestamp: (/* @__PURE__ */ new Date()).toLocaleTimeString(),
      type: "system"
    }
  ]);
  const [input, setInput] = useState("");
  const [isReasoning, setIsReasoning] = useState(false);
  const [reasoningLog, setReasoningLog] = useState(null);
  const scrollRef = useRef(null);
  useEffect(() => {
    if (scrollRef.current) {
      scrollRef.current.scrollTop = scrollRef.current.scrollHeight;
    }
  }, [messages]);
  const handleSend = async () => {
    if (!input.trim()) return;
    const userMessage = {
      role: "user",
      content: input,
      timestamp: (/* @__PURE__ */ new Date()).toLocaleTimeString()
    };
    setMessages((prev) => [...prev, userMessage]);
    setInput("");
    setIsReasoning(true);
    try {
      const result = onSend ? await onSend(input) : {
        summary: "Local mode \u2014 connect an AI provider for full reasoning.",
        threats: [],
        recommendations: ["Configure AI provider in environment"],
        threatLevel: 0,
        reasoning: "No AI provider connected."
      };
      setReasoningLog(result);
      const agentMessage = {
        role: "agent",
        content: result.summary,
        timestamp: (/* @__PURE__ */ new Date()).toLocaleTimeString(),
        recommendations: result.recommendations,
        threatLevel: result.threatLevel
      };
      setMessages((prev) => [...prev, agentMessage]);
    } catch (error) {
      setMessages((prev) => [
        ...prev,
        {
          role: "agent",
          content: "Error in reasoning engine.",
          timestamp: (/* @__PURE__ */ new Date()).toLocaleTimeString(),
          type: "system"
        }
      ]);
    } finally {
      setIsReasoning(false);
    }
  };
  const modeConfig = {
    security: { title: "AI Security Agent", subtitle: "LLM-BASED REASONING ENGINE v2.0", badge: "Autonomous Mode", badgeColor: "text-emerald-500" },
    general: { title: "AI Assistant", subtitle: "REASONING ENGINE", badge: "Active", badgeColor: "text-blue-500" },
    oversight: { title: "Oversight Agent", subtitle: "HUMAN-IN-THE-LOOP MONITOR", badge: "Supervising", badgeColor: "text-amber-500" }
  };
  const config = modeConfig[mode];
  return /* @__PURE__ */ jsxs4("div", { className: "grid grid-cols-1 lg:grid-cols-3 gap-6 h-[calc(100vh-12rem)]", children: [
    /* @__PURE__ */ jsxs4(Card, { className: "lg:col-span-2 flex flex-col", children: [
      /* @__PURE__ */ jsx15(CardHeader, { className: "border-b", children: /* @__PURE__ */ jsxs4("div", { className: "flex items-center justify-between", children: [
        /* @__PURE__ */ jsxs4("div", { className: "flex items-center gap-3", children: [
          /* @__PURE__ */ jsx15("div", { className: "p-2 rounded-lg bg-primary/10", children: /* @__PURE__ */ jsx15(Brain, { className: "w-5 h-5 text-primary" }) }),
          /* @__PURE__ */ jsxs4("div", { children: [
            /* @__PURE__ */ jsx15(CardTitle, { className: "text-lg", children: config.title }),
            /* @__PURE__ */ jsx15(CardDescription, { className: "text-xs font-mono", children: config.subtitle })
          ] })
        ] }),
        /* @__PURE__ */ jsx15(Badge, { variant: "outline", className: `font-mono text-[10px] uppercase tracking-widest border-emerald-500/30 ${config.badgeColor}`, children: config.badge })
      ] }) }),
      /* @__PURE__ */ jsx15(CardContent, { className: "flex-1 p-0 overflow-hidden", children: /* @__PURE__ */ jsx15(ScrollArea, { className: "h-full p-6", ref: scrollRef, children: /* @__PURE__ */ jsxs4("div", { className: "space-y-6", children: [
        messages.map((msg, i) => /* @__PURE__ */ jsx15(
          motion.div,
          {
            initial: { opacity: 0, x: msg.role === "user" ? 10 : -10 },
            animate: { opacity: 1, x: 0 },
            className: `flex ${msg.role === "user" ? "justify-end" : "justify-start"}`,
            children: /* @__PURE__ */ jsxs4("div", { className: `max-w-[80%] space-y-2 ${msg.role === "user" ? "items-end" : "items-start"}`, children: [
              /* @__PURE__ */ jsxs4(
                "div",
                {
                  className: `p-4 rounded-2xl text-sm ${msg.role === "user" ? "bg-primary text-primary-foreground rounded-tr-none" : "bg-muted/50 border rounded-tl-none"}`,
                  children: [
                    msg.content,
                    msg.recommendations && /* @__PURE__ */ jsxs4("div", { className: "mt-4 space-y-2", children: [
                      /* @__PURE__ */ jsx15("p", { className: "text-[10px] font-bold uppercase tracking-widest opacity-70", children: "Recommended Actions:" }),
                      msg.recommendations.map((rec, j) => /* @__PURE__ */ jsxs4("div", { className: "flex items-start gap-2 text-xs bg-background/30 p-2 rounded-lg border border-white/5", children: [
                        /* @__PURE__ */ jsx15(ChevronRight, { className: "w-3 h-3 mt-0.5 text-primary shrink-0" }),
                        /* @__PURE__ */ jsx15("span", { children: rec })
                      ] }, j))
                    ] })
                  ]
                }
              ),
              /* @__PURE__ */ jsx15("p", { className: "text-[10px] text-muted-foreground font-mono px-1", children: msg.timestamp })
            ] })
          },
          i
        )),
        isReasoning && /* @__PURE__ */ jsx15("div", { className: "flex justify-start", children: /* @__PURE__ */ jsxs4("div", { className: "bg-muted/30 border p-4 rounded-2xl rounded-tl-none flex items-center gap-3", children: [
          /* @__PURE__ */ jsx15(Loader2, { className: "w-4 h-4 animate-spin text-primary" }),
          /* @__PURE__ */ jsx15("span", { className: "text-xs font-mono animate-pulse", children: "Agent is reasoning..." })
        ] }) })
      ] }) }) }),
      /* @__PURE__ */ jsx15(CardFooter, { className: "p-4 border-t", children: /* @__PURE__ */ jsxs4("div", { className: "flex w-full gap-3", children: [
        /* @__PURE__ */ jsx15(
          Input,
          {
            placeholder: "Ask the agent...",
            value: input,
            onChange: (e) => setInput(e.target.value),
            onKeyDown: (e) => e.key === "Enter" && handleSend(),
            className: "bg-background/50"
          }
        ),
        /* @__PURE__ */ jsx15(Button, { onClick: handleSend, disabled: isReasoning || !input.trim(), children: /* @__PURE__ */ jsx15(Send, { className: "w-4 h-4" }) })
      ] }) })
    ] }),
    /* @__PURE__ */ jsxs4("div", { className: "space-y-6", children: [
      /* @__PURE__ */ jsxs4(Card, { children: [
        /* @__PURE__ */ jsx15(CardHeader, { className: "pb-2", children: /* @__PURE__ */ jsxs4("div", { className: "flex items-center gap-2", children: [
          /* @__PURE__ */ jsx15(Terminal, { className: "w-4 h-4 text-primary" }),
          /* @__PURE__ */ jsx15(CardTitle, { className: "text-sm font-mono uppercase tracking-widest", children: "Reasoning Log" })
        ] }) }),
        /* @__PURE__ */ jsx15(CardContent, { className: "space-y-4", children: reasoningLog ? /* @__PURE__ */ jsxs4("div", { className: "space-y-4", children: [
          /* @__PURE__ */ jsxs4("div", { className: "p-3 rounded-lg bg-background/50 border font-mono text-[11px] leading-relaxed", children: [
            /* @__PURE__ */ jsx15("span", { className: "text-primary font-bold", children: "LOG_ENTRY:" }),
            " ",
            reasoningLog.reasoning
          ] }),
          /* @__PURE__ */ jsxs4("div", { className: "grid grid-cols-2 gap-3", children: [
            /* @__PURE__ */ jsxs4("div", { className: "p-3 rounded-lg bg-background/50 border text-center", children: [
              /* @__PURE__ */ jsx15("p", { className: "text-[10px] text-muted-foreground uppercase font-mono mb-1", children: "Threat Level" }),
              /* @__PURE__ */ jsx15("p", { className: `text-xl font-bold ${reasoningLog.threatLevel > 50 ? "text-destructive" : "text-emerald-500"}`, children: reasoningLog.threatLevel })
            ] }),
            /* @__PURE__ */ jsxs4("div", { className: "p-3 rounded-lg bg-background/50 border text-center", children: [
              /* @__PURE__ */ jsx15("p", { className: "text-[10px] text-muted-foreground uppercase font-mono mb-1", children: "Confidence" }),
              /* @__PURE__ */ jsx15("p", { className: "text-xl font-bold text-primary", children: "94%" })
            ] })
          ] })
        ] }) : /* @__PURE__ */ jsxs4("div", { className: "h-32 flex flex-col items-center justify-center text-muted-foreground opacity-50 border-2 border-dashed rounded-xl", children: [
          /* @__PURE__ */ jsx15(Cpu, { className: "w-8 h-8 mb-2" }),
          /* @__PURE__ */ jsx15("p", { className: "text-xs font-mono", children: "Waiting for inference..." })
        ] }) })
      ] }),
      /* @__PURE__ */ jsxs4(Card, { children: [
        /* @__PURE__ */ jsx15(CardHeader, { className: "pb-2", children: /* @__PURE__ */ jsx15(CardTitle, { className: "text-sm font-semibold", children: "Active Guardrails" }) }),
        /* @__PURE__ */ jsxs4(CardContent, { className: "space-y-3", children: [
          /* @__PURE__ */ jsx15(GuardrailItem, { icon: CheckCircle2, label: "Explainable Decisions", status: "Active", color: "text-emerald-500" }),
          /* @__PURE__ */ jsx15(GuardrailItem, { icon: ShieldAlert, label: "Human-in-the-loop", status: "Threshold: High", color: "text-amber-500" }),
          /* @__PURE__ */ jsx15(GuardrailItem, { icon: Info, label: "Immutable Audit Trail", status: "Encrypted", color: "text-blue-500" })
        ] })
      ] })
    ] })
  ] });
}
function GuardrailItem({ icon: Icon2, label, status, color }) {
  return /* @__PURE__ */ jsxs4("div", { className: "flex items-center justify-between p-2 rounded-lg bg-muted/30 border border-border/20", children: [
    /* @__PURE__ */ jsxs4("div", { className: "flex items-center gap-2", children: [
      /* @__PURE__ */ jsx15(Icon2, { className: `w-3.5 h-3.5 ${color}` }),
      /* @__PURE__ */ jsx15("span", { className: "text-xs font-medium", children: label })
    ] }),
    /* @__PURE__ */ jsx15("span", { className: "text-[10px] font-mono text-muted-foreground", children: status })
  ] });
}

// src/components/ai/AgentFleet.tsx
import { useState as useState2 } from "react";
import { Users, Cpu as Cpu2, Zap, Activity, Network, Bot, Terminal as TerminalIcon, CheckCircle2 as CheckCircle22, Loader2 as Loader22, Plus, AlertTriangle } from "lucide-react";
import { jsx as jsx16, jsxs as jsxs5 } from "react/jsx-runtime";
function AgentFleet({ workers: initialWorkers, onSpawnWorker, onViewNodeGraph }) {
  const [workers, setWorkers] = useState2(
    initialWorkers || [
      { id: "1", name: "MarketScraper_01", task: "Scraping tech crunch for funding news", status: "working", load: 78 },
      { id: "2", name: "LeadEnricher_01", task: "Awaiting leads from Scraper", status: "idle", load: 0 },
      { id: "3", name: "ContentDraft_01", task: "Generating social media copy", status: "working", load: 45 },
      { id: "4", name: "FinanceAuditor_01", task: "Verifying daily revenue sync", status: "complete", load: 0 }
    ]
  );
  const handleSpawn = () => {
    if (onSpawnWorker) {
      onSpawnWorker();
      return;
    }
    const newWorker = {
      id: Math.random().toString(36).slice(2, 8),
      name: `Worker_${Math.floor(Math.random() * 1e3)}`,
      task: "Initializing strategic analysis...",
      status: "idle",
      load: 0
    };
    setWorkers((prev) => [...prev, newWorker]);
  };
  const activeCount = workers.filter((w) => w.status === "working").length;
  const awaitingApproval = workers.filter((w) => w.status === "awaiting_approval").length;
  const errorCount = workers.filter((w) => w.status === "error").length;
  const avgLoad = workers.length > 0 ? Math.round(workers.reduce((sum, w) => sum + w.load, 0) / workers.length) : 0;
  return /* @__PURE__ */ jsxs5("div", { className: "h-full p-8 space-y-8", children: [
    /* @__PURE__ */ jsxs5("div", { className: "flex items-center justify-between border-b pb-6", children: [
      /* @__PURE__ */ jsxs5("div", { children: [
        /* @__PURE__ */ jsxs5("h2", { className: "text-2xl font-black text-emerald-500 tracking-tighter flex items-center gap-3 italic", children: [
          /* @__PURE__ */ jsx16(Network, { className: "h-8 w-8" }),
          "MULTI-AGENT ORCHESTRATOR"
        ] }),
        /* @__PURE__ */ jsx16("p", { className: "text-muted-foreground text-xs font-bold uppercase tracking-widest mt-1", children: "Parallel reasoning & distributed task execution" })
      ] }),
      /* @__PURE__ */ jsxs5("div", { className: "flex gap-2", children: [
        /* @__PURE__ */ jsx16(Button, { variant: "outline", onClick: onViewNodeGraph, children: "View Node Graph" }),
        /* @__PURE__ */ jsxs5(Button, { onClick: handleSpawn, className: "bg-emerald-500 hover:bg-emerald-600 text-black font-black uppercase text-[10px] tracking-widest", children: [
          "Spawn Worker ",
          /* @__PURE__ */ jsx16(Plus, { className: "h-3 w-3 ml-2" })
        ] })
      ] })
    ] }),
    /* @__PURE__ */ jsxs5("div", { className: "grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6", children: [
      /* @__PURE__ */ jsxs5(Card, { className: "border-emerald-500/30 border-2 relative overflow-hidden group", children: [
        /* @__PURE__ */ jsx16("div", { className: "absolute top-0 right-0 p-8 opacity-5 group-hover:scale-110 transition-transform", children: /* @__PURE__ */ jsx16(Cpu2, { className: "h-24 w-24 text-emerald-500" }) }),
        /* @__PURE__ */ jsxs5(CardHeader, { children: [
          /* @__PURE__ */ jsxs5("div", { className: "flex justify-between items-start", children: [
            /* @__PURE__ */ jsxs5("div", { className: "flex items-center gap-3", children: [
              /* @__PURE__ */ jsx16("div", { className: "p-2 bg-emerald-500/10 rounded-lg", children: /* @__PURE__ */ jsx16(Zap, { className: "h-5 w-5 text-emerald-400" }) }),
              /* @__PURE__ */ jsx16(CardTitle, { className: "text-base italic uppercase tracking-widest", children: "Master Orchestrator" })
            ] }),
            /* @__PURE__ */ jsx16(Badge, { className: "bg-emerald-500 text-black animate-pulse", children: "OPTIMIZED" })
          ] }),
          /* @__PURE__ */ jsx16(CardDescription, { className: "text-muted-foreground text-[10px] uppercase font-black tracking-widest mt-2", children: "Core Coordination Logic" })
        ] }),
        /* @__PURE__ */ jsxs5(CardContent, { className: "space-y-6", children: [
          /* @__PURE__ */ jsxs5("div", { className: "space-y-3", children: [
            /* @__PURE__ */ jsxs5("div", { className: "flex justify-between text-[10px] font-bold uppercase italic", children: [
              /* @__PURE__ */ jsx16("span", { className: "text-muted-foreground", children: "GLOBAL COORDINATION LOAD" }),
              /* @__PURE__ */ jsxs5("span", { className: "text-emerald-400", children: [
                avgLoad,
                "%"
              ] })
            ] }),
            /* @__PURE__ */ jsx16("div", { className: "h-1.5 w-full bg-muted rounded-full overflow-hidden", children: /* @__PURE__ */ jsx16("div", { className: "h-full bg-emerald-500 transition-all", style: { width: `${avgLoad}%` } }) })
          ] }),
          /* @__PURE__ */ jsxs5("div", { className: "grid grid-cols-2 gap-3", children: [
            /* @__PURE__ */ jsxs5("div", { className: "p-3 bg-muted rounded-lg text-center", children: [
              /* @__PURE__ */ jsx16("p", { className: "text-xs font-black text-white", children: activeCount }),
              /* @__PURE__ */ jsx16("p", { className: "text-[8px] text-muted-foreground uppercase font-black", children: "ACTIVE AGENTS" })
            ] }),
            /* @__PURE__ */ jsxs5("div", { className: "p-3 bg-muted rounded-lg text-center", children: [
              /* @__PURE__ */ jsx16("p", { className: "text-xs font-black text-white", children: workers.length }),
              /* @__PURE__ */ jsx16("p", { className: "text-[8px] text-muted-foreground uppercase font-black", children: "TOTAL WORKERS" })
            ] }),
            awaitingApproval > 0 && /* @__PURE__ */ jsxs5("div", { className: "p-3 bg-amber-500/10 border border-amber-500/30 rounded-lg text-center", children: [
              /* @__PURE__ */ jsx16("p", { className: "text-xs font-black text-amber-500", children: awaitingApproval }),
              /* @__PURE__ */ jsx16("p", { className: "text-[8px] text-amber-500/70 uppercase font-black", children: "AWAITING APPROVAL" })
            ] }),
            errorCount > 0 && /* @__PURE__ */ jsxs5("div", { className: "p-3 bg-destructive/10 border border-destructive/30 rounded-lg text-center", children: [
              /* @__PURE__ */ jsx16("p", { className: "text-xs font-black text-destructive", children: errorCount }),
              /* @__PURE__ */ jsx16("p", { className: "text-[8px] text-destructive/70 uppercase font-black", children: "ERRORS" })
            ] })
          ] })
        ] })
      ] }),
      /* @__PURE__ */ jsxs5("div", { className: "col-span-1 md:col-span-2 space-y-4", children: [
        /* @__PURE__ */ jsxs5("h3", { className: "text-xs font-black uppercase text-muted-foreground tracking-[0.2em] flex items-center gap-2", children: [
          /* @__PURE__ */ jsx16(Activity, { className: "h-3 w-3" }),
          " Worker Fleet Status"
        ] }),
        /* @__PURE__ */ jsx16("div", { className: "grid grid-cols-1 sm:grid-cols-2 gap-4", children: workers.map((worker) => /* @__PURE__ */ jsxs5(
          "div",
          {
            className: "p-4 bg-card border rounded-xl hover:border-border transition-all flex items-center justify-between group cursor-pointer",
            children: [
              /* @__PURE__ */ jsxs5("div", { className: "flex items-center gap-4", children: [
                /* @__PURE__ */ jsxs5(
                  "div",
                  {
                    className: cn(
                      "h-10 w-10 rounded-lg flex items-center justify-center transition-colors",
                      worker.status === "working" && "bg-emerald-500/10 text-emerald-500",
                      worker.status === "complete" && "bg-blue-500/10 text-blue-500",
                      worker.status === "error" && "bg-destructive/10 text-destructive",
                      worker.status === "awaiting_approval" && "bg-amber-500/10 text-amber-500",
                      worker.status === "idle" && "bg-muted text-muted-foreground"
                    ),
                    children: [
                      worker.status === "working" && /* @__PURE__ */ jsx16(Loader22, { className: "h-5 w-5 animate-spin" }),
                      worker.status === "complete" && /* @__PURE__ */ jsx16(CheckCircle22, { className: "h-5 w-5" }),
                      worker.status === "error" && /* @__PURE__ */ jsx16(AlertTriangle, { className: "h-5 w-5" }),
                      worker.status === "awaiting_approval" && /* @__PURE__ */ jsx16(Users, { className: "h-5 w-5" }),
                      worker.status === "idle" && /* @__PURE__ */ jsx16(Bot, { className: "h-5 w-5" })
                    ]
                  }
                ),
                /* @__PURE__ */ jsxs5("div", { className: "space-y-0.5", children: [
                  /* @__PURE__ */ jsx16("p", { className: "text-[11px] font-black text-foreground uppercase tracking-tight", children: worker.name }),
                  /* @__PURE__ */ jsx16("p", { className: "text-[10px] text-muted-foreground line-clamp-1 italic", children: worker.task })
                ] })
              ] }),
              /* @__PURE__ */ jsxs5("div", { className: "text-right", children: [
                /* @__PURE__ */ jsxs5("p", { className: "text-[10px] font-black text-emerald-500", children: [
                  worker.load,
                  "%"
                ] }),
                /* @__PURE__ */ jsx16("div", { className: "h-1 w-12 bg-muted rounded-full mt-1", children: /* @__PURE__ */ jsx16("div", { className: "h-full bg-emerald-500/50 transition-all", style: { width: `${worker.load}%` } }) })
              ] })
            ]
          },
          worker.id
        )) })
      ] })
    ] }),
    /* @__PURE__ */ jsxs5(Card, { children: [
      /* @__PURE__ */ jsxs5("div", { className: "h-8 border-b flex items-center px-4", children: [
        /* @__PURE__ */ jsx16(TerminalIcon, { className: "h-3 w-3 text-emerald-500 mr-2" }),
        /* @__PURE__ */ jsx16("span", { className: "text-[9px] font-black text-muted-foreground uppercase tracking-[0.2em]", children: "Distributed Logic Logs" })
      ] }),
      /* @__PURE__ */ jsxs5("div", { className: "p-6 font-mono text-[10px] space-y-2 text-muted-foreground leading-relaxed italic", children: [
        /* @__PURE__ */ jsxs5("div", { className: "flex gap-4", children: [
          /* @__PURE__ */ jsx16("span", { className: "text-muted-foreground/30 shrink-0", children: (/* @__PURE__ */ new Date()).toLocaleTimeString() }),
          /* @__PURE__ */ jsx16("span", { children: "[MASTER] Orchestrator initialized. Fleet ready." })
        ] }),
        workers.filter((w) => w.status === "working").map((w) => /* @__PURE__ */ jsxs5("div", { className: "flex gap-4", children: [
          /* @__PURE__ */ jsx16("span", { className: "text-muted-foreground/30 shrink-0", children: (/* @__PURE__ */ new Date()).toLocaleTimeString() }),
          /* @__PURE__ */ jsxs5("span", { className: "text-emerald-500", children: [
            "[",
            w.name,
            "] ",
            w.task
          ] })
        ] }, w.id)),
        workers.filter((w) => w.status === "awaiting_approval").map((w) => /* @__PURE__ */ jsxs5("div", { className: "flex gap-4", children: [
          /* @__PURE__ */ jsx16("span", { className: "text-muted-foreground/30 shrink-0", children: (/* @__PURE__ */ new Date()).toLocaleTimeString() }),
          /* @__PURE__ */ jsxs5("span", { className: "text-amber-500", children: [
            "[",
            w.name,
            "] Awaiting human approval..."
          ] })
        ] }, w.id))
      ] })
    ] })
  ] });
}

// src/components/governance/GovernancePanel.tsx
import { useState as useState3 } from "react";
import { Shield, ShieldAlert as ShieldAlert2, ShieldCheck, Lock, Eye, AlertTriangle as AlertTriangle2, Scale } from "lucide-react";
import { jsx as jsx17, jsxs as jsxs6 } from "react/jsx-runtime";
var DEFAULT_POLICIES = [
  {
    id: "POL-001",
    name: "Budget Gate",
    description: "Prevents any agent action exceeding $500 without human approval. Applies to all financial operations.",
    status: "active",
    tier: "CRITICAL"
  },
  {
    id: "POL-002",
    name: "Financial Reconciliation",
    description: "All payment and invoice operations must be verified against Stripe before execution.",
    status: "active",
    tier: "CRITICAL"
  },
  {
    id: "POL-003",
    name: "Code Sandbox",
    description: "Generated code must pass dry-run in isolated sandbox before deployment. No direct production writes.",
    status: "active",
    tier: "HIGH"
  },
  {
    id: "POL-004",
    name: "PII Redaction",
    description: "All prompts are scanned for emails, phone numbers, and SSNs. Detected PII is redacted before LLM processing.",
    status: "active",
    tier: "HIGH"
  },
  {
    id: "POL-005",
    name: "Deployment Guard",
    description: "Production deployments require passing all preflight checks: lint, test, security scan, bundle size.",
    status: "active",
    tier: "HIGH"
  },
  {
    id: "POL-006",
    name: "External API Rate Limit",
    description: "Agent scraping and external API calls are rate-limited to prevent abuse and IP bans.",
    status: "active",
    tier: "MEDIUM"
  },
  {
    id: "POL-007",
    name: "Auth Boundary",
    description: "Agents cannot access or modify authentication tokens, session cookies, or user credentials.",
    status: "active",
    tier: "CRITICAL"
  }
];
var ICON_MAP = {
  "POL-001": Lock,
  "POL-002": Scale,
  "POL-003": ShieldCheck,
  "POL-004": Eye,
  "POL-005": ShieldAlert2,
  "POL-006": AlertTriangle2,
  "POL-007": Lock
};
var TIER_COLORS = {
  CRITICAL: "text-destructive border-destructive/30 bg-destructive/5",
  HIGH: "text-amber-500 border-amber-500/30 bg-amber-500/5",
  MEDIUM: "text-blue-500 border-blue-500/30 bg-blue-500/5",
  LOW: "text-muted-foreground border-border bg-muted/50"
};
function GovernancePanel({ policies: initialPolicies, onToggle }) {
  const [policies, setPolicies] = useState3(initialPolicies || DEFAULT_POLICIES);
  const handleToggle = (id, currentStatus) => {
    const newStatus = currentStatus === "active" ? "inactive" : "active";
    setPolicies((prev) => prev.map((p) => p.id === id ? { ...p, status: newStatus } : p));
    onToggle?.(id, newStatus);
  };
  const activeCount = policies.filter((p) => p.status === "active").length;
  const criticalActive = policies.filter((p) => p.tier === "CRITICAL" && p.status === "active").length;
  return /* @__PURE__ */ jsxs6("div", { className: "flex-1 flex flex-col p-6 overflow-hidden", children: [
    /* @__PURE__ */ jsxs6("div", { className: "flex items-center justify-between mb-8", children: [
      /* @__PURE__ */ jsxs6("div", { children: [
        /* @__PURE__ */ jsxs6("h2", { className: "text-xl font-bold flex items-center gap-2", children: [
          /* @__PURE__ */ jsx17(Shield, { className: "h-5 w-5 text-amber-500" }),
          "Agent Governance Engine"
        ] }),
        /* @__PURE__ */ jsx17("p", { className: "text-sm text-muted-foreground mt-1", children: "Machine-enforced boundaries and trust policies." })
      ] }),
      /* @__PURE__ */ jsxs6("div", { className: "flex items-center gap-3", children: [
        /* @__PURE__ */ jsxs6(Badge, { variant: "outline", className: "text-[10px] py-0 border-amber-500/30 text-amber-500 bg-amber-500/5", children: [
          activeCount,
          "/",
          policies.length,
          " ENFORCED"
        ] }),
        criticalActive < policies.filter((p) => p.tier === "CRITICAL").length && /* @__PURE__ */ jsx17(Badge, { variant: "outline", className: "text-[10px] py-0 border-destructive/30 text-destructive bg-destructive/5", children: "CRITICAL GAPS" })
      ] })
    ] }),
    /* @__PURE__ */ jsx17(ScrollArea, { className: "flex-1", children: /* @__PURE__ */ jsx17("div", { className: "grid grid-cols-1 gap-4 pb-8", children: policies.map((policy) => {
      const Icon2 = ICON_MAP[policy.id] || Shield;
      return /* @__PURE__ */ jsx17(Card, { className: "hover:border-border/80 transition-colors", children: /* @__PURE__ */ jsxs6(CardContent, { className: "p-4 flex items-start justify-between", children: [
        /* @__PURE__ */ jsxs6("div", { className: "flex gap-4", children: [
          /* @__PURE__ */ jsx17("div", { className: "h-10 w-10 rounded-lg bg-muted border flex items-center justify-center shrink-0", children: /* @__PURE__ */ jsx17(Icon2, { className: `h-5 w-5 ${policy.status === "active" ? "text-amber-500" : "text-muted-foreground/50"}` }) }),
          /* @__PURE__ */ jsxs6("div", { children: [
            /* @__PURE__ */ jsxs6("div", { className: "flex items-center gap-2 mb-1", children: [
              /* @__PURE__ */ jsx17("h3", { className: "font-bold text-sm", children: policy.name }),
              /* @__PURE__ */ jsx17(Badge, { variant: "outline", className: `text-[10px] py-0 ${TIER_COLORS[policy.tier]}`, children: policy.tier })
            ] }),
            /* @__PURE__ */ jsx17("p", { className: "text-xs text-muted-foreground max-w-md", children: policy.description }),
            /* @__PURE__ */ jsxs6("div", { className: "mt-3 flex items-center gap-4 text-[10px] font-mono text-muted-foreground", children: [
              /* @__PURE__ */ jsxs6("span", { children: [
                "ID: ",
                policy.id
              ] }),
              /* @__PURE__ */ jsxs6("span", { className: "flex items-center gap-1", children: [
                /* @__PURE__ */ jsx17("span", { className: `inline-block w-1.5 h-1.5 rounded-full ${policy.status === "active" ? "bg-green-500" : "bg-red-500"}` }),
                "STATE: ",
                policy.status === "active" ? "ENFORCED" : "BYPASSED"
              ] })
            ] })
          ] })
        ] }),
        /* @__PURE__ */ jsx17("div", { className: "flex flex-col items-end gap-2", children: /* @__PURE__ */ jsx17(Switch, { checked: policy.status === "active", onCheckedChange: () => handleToggle(policy.id, policy.status) }) })
      ] }) }, policy.id);
    }) }) })
  ] });
}

// src/components/governance/OversightPanel.tsx
import { useState as useState4 } from "react";
import { Eye as Eye2, Shield as Shield2, AlertTriangle as AlertTriangle3, CheckCircle, XCircle, Clock, Activity as Activity2 } from "lucide-react";
import { motion as motion2, AnimatePresence } from "motion/react";
import { jsx as jsx18, jsxs as jsxs7 } from "react/jsx-runtime";
var DEFAULT_EVENTS = [
  {
    id: "ESC-001",
    agent: "FinanceAuditor_01",
    action: "Process refund of $5,200",
    reason: "Value threshold exceeded ($5K limit)",
    confidence: 0.92,
    timestamp: "18:22:01",
    mode: "checkpoint",
    status: "pending",
    risk: "high"
  },
  {
    id: "ESC-002",
    agent: "ContentDraft_01",
    action: "Publish marketing email to 12,000 contacts",
    reason: "Mass communication requires human review",
    confidence: 0.78,
    timestamp: "18:23:15",
    mode: "shadow",
    status: "pending",
    risk: "medium"
  },
  {
    id: "ESC-003",
    agent: "MarketScraper_01",
    action: "Scrape competitor pricing data",
    reason: "External data collection",
    confidence: 0.95,
    timestamp: "18:24:00",
    mode: "shadow",
    status: "approved",
    risk: "low"
  },
  {
    id: "ESC-004",
    agent: "LeadEnricher_01",
    action: "Update CRM with 340 new leads",
    reason: "Bulk data modification",
    confidence: 0.65,
    timestamp: "18:25:30",
    mode: "checkpoint",
    status: "pending",
    risk: "medium"
  }
];
var RISK_COLORS = {
  low: "text-emerald-500 bg-emerald-500/10 border-emerald-500/30",
  medium: "text-amber-500 bg-amber-500/10 border-amber-500/30",
  high: "text-orange-500 bg-orange-500/10 border-orange-500/30",
  critical: "text-destructive bg-destructive/10 border-destructive/30"
};
var MODE_CONFIG = {
  shadow: { label: "SHADOW", icon: Eye2, description: "Agents act unless flagged" },
  checkpoint: { label: "CHECKPOINT", icon: Shield2, description: "Agents pause for approval" },
  recovery: { label: "RECOVERY", icon: AlertTriangle3, description: "Humans take control" }
};
function OversightPanel({
  events: initialEvents,
  onApprove,
  onReject,
  mode = "checkpoint",
  onModeChange
}) {
  const [events, setEvents] = useState4(initialEvents || DEFAULT_EVENTS);
  const pendingCount = events.filter((e) => e.status === "pending").length;
  const approvedCount = events.filter((e) => e.status === "approved").length;
  const rejectedCount = events.filter((e) => e.status === "rejected").length;
  const handleApprove = (id) => {
    setEvents((prev) => prev.map((e) => e.id === id ? { ...e, status: "approved" } : e));
    onApprove?.(id);
  };
  const handleReject = (id) => {
    setEvents((prev) => prev.map((e) => e.id === id ? { ...e, status: "rejected" } : e));
    onReject?.(id);
  };
  const ModeIcon = MODE_CONFIG[mode].icon;
  return /* @__PURE__ */ jsxs7("div", { className: "h-full p-6 space-y-6", children: [
    /* @__PURE__ */ jsxs7("div", { className: "flex items-center justify-between", children: [
      /* @__PURE__ */ jsxs7("div", { children: [
        /* @__PURE__ */ jsxs7("h2", { className: "text-2xl font-black text-amber-500 tracking-tighter flex items-center gap-3 italic", children: [
          /* @__PURE__ */ jsx18(Eye2, { className: "h-8 w-8" }),
          "HUMAN OVERSIGHT"
        ] }),
        /* @__PURE__ */ jsx18("p", { className: "text-muted-foreground text-xs font-bold uppercase tracking-widest mt-1", children: "AI supervisor dashboard \u2014 approve, reject, escalate" })
      ] }),
      /* @__PURE__ */ jsx18("div", { className: "flex gap-2", children: Object.keys(MODE_CONFIG).map((m) => {
        const C = MODE_CONFIG[m];
        return /* @__PURE__ */ jsxs7(
          Button,
          {
            variant: mode === m ? "default" : "outline",
            size: "sm",
            onClick: () => onModeChange?.(m),
            className: mode === m ? "bg-amber-500 text-black hover:bg-amber-600" : "",
            children: [
              /* @__PURE__ */ jsx18(C.icon, { className: "h-3 w-3 mr-1" }),
              C.label
            ]
          },
          m
        );
      }) })
    ] }),
    /* @__PURE__ */ jsxs7("div", { className: "grid grid-cols-4 gap-4", children: [
      /* @__PURE__ */ jsx18(Card, { children: /* @__PURE__ */ jsxs7(CardContent, { className: "p-4 flex items-center gap-3", children: [
        /* @__PURE__ */ jsx18(Clock, { className: "h-8 w-8 text-amber-500" }),
        /* @__PURE__ */ jsxs7("div", { children: [
          /* @__PURE__ */ jsx18("p", { className: "text-2xl font-black", children: pendingCount }),
          /* @__PURE__ */ jsx18("p", { className: "text-[10px] text-muted-foreground uppercase font-black", children: "PENDING" })
        ] })
      ] }) }),
      /* @__PURE__ */ jsx18(Card, { children: /* @__PURE__ */ jsxs7(CardContent, { className: "p-4 flex items-center gap-3", children: [
        /* @__PURE__ */ jsx18(CheckCircle, { className: "h-8 w-8 text-emerald-500" }),
        /* @__PURE__ */ jsxs7("div", { children: [
          /* @__PURE__ */ jsx18("p", { className: "text-2xl font-black", children: approvedCount }),
          /* @__PURE__ */ jsx18("p", { className: "text-[10px] text-muted-foreground uppercase font-black", children: "APPROVED" })
        ] })
      ] }) }),
      /* @__PURE__ */ jsx18(Card, { children: /* @__PURE__ */ jsxs7(CardContent, { className: "p-4 flex items-center gap-3", children: [
        /* @__PURE__ */ jsx18(XCircle, { className: "h-8 w-8 text-destructive" }),
        /* @__PURE__ */ jsxs7("div", { children: [
          /* @__PURE__ */ jsx18("p", { className: "text-2xl font-black", children: rejectedCount }),
          /* @__PURE__ */ jsx18("p", { className: "text-[10px] text-muted-foreground uppercase font-black", children: "REJECTED" })
        ] })
      ] }) }),
      /* @__PURE__ */ jsx18(Card, { children: /* @__PURE__ */ jsxs7(CardContent, { className: "p-4 flex items-center gap-3", children: [
        /* @__PURE__ */ jsx18(Activity2, { className: "h-8 w-8 text-blue-500" }),
        /* @__PURE__ */ jsxs7("div", { children: [
          /* @__PURE__ */ jsx18("p", { className: "text-2xl font-black", children: events.length }),
          /* @__PURE__ */ jsx18("p", { className: "text-[10px] text-muted-foreground uppercase font-black", children: "TOTAL" })
        ] })
      ] }) })
    ] }),
    /* @__PURE__ */ jsxs7(Card, { children: [
      /* @__PURE__ */ jsx18(CardHeader, { className: "pb-3", children: /* @__PURE__ */ jsxs7("div", { className: "flex items-center justify-between", children: [
        /* @__PURE__ */ jsxs7("div", { children: [
          /* @__PURE__ */ jsxs7(CardTitle, { className: "flex items-center gap-2", children: [
            /* @__PURE__ */ jsx18(ModeIcon, { className: "h-4 w-4 text-amber-500" }),
            MODE_CONFIG[mode].label,
            " MODE"
          ] }),
          /* @__PURE__ */ jsx18(CardDescription, { children: MODE_CONFIG[mode].description })
        ] }),
        /* @__PURE__ */ jsx18(Badge, { variant: "outline", className: "text-[10px] py-0 border-amber-500/30 text-amber-500 bg-amber-500/5", children: MODE_CONFIG[mode].label })
      ] }) }),
      /* @__PURE__ */ jsx18(CardContent, { children: /* @__PURE__ */ jsx18(ScrollArea, { className: "h-[400px] pr-4", children: /* @__PURE__ */ jsx18("div", { className: "space-y-4", children: /* @__PURE__ */ jsx18(AnimatePresence, { children: events.map((event) => /* @__PURE__ */ jsx18(
        motion2.div,
        {
          initial: { opacity: 0, y: 10 },
          animate: { opacity: 1, y: 0 },
          exit: { opacity: 0, x: -20 },
          children: /* @__PURE__ */ jsx18(Card, { className: `border ${event.status === "pending" ? "border-amber-500/30" : "border-border/50"}`, children: /* @__PURE__ */ jsx18(CardContent, { className: "p-4", children: /* @__PURE__ */ jsxs7("div", { className: "flex items-start justify-between", children: [
            /* @__PURE__ */ jsxs7("div", { className: "space-y-2 flex-1", children: [
              /* @__PURE__ */ jsxs7("div", { className: "flex items-center gap-2", children: [
                /* @__PURE__ */ jsx18(Badge, { variant: "outline", className: `text-[10px] py-0 ${RISK_COLORS[event.risk]}`, children: event.risk.toUpperCase() }),
                /* @__PURE__ */ jsx18("span", { className: "text-[10px] font-mono text-muted-foreground", children: event.id }),
                /* @__PURE__ */ jsx18("span", { className: "text-[10px] font-mono text-muted-foreground", children: event.timestamp })
              ] }),
              /* @__PURE__ */ jsxs7("div", { children: [
                /* @__PURE__ */ jsx18("p", { className: "text-sm font-bold", children: event.action }),
                /* @__PURE__ */ jsxs7("p", { className: "text-xs text-muted-foreground mt-1", children: [
                  /* @__PURE__ */ jsx18("span", { className: "font-mono text-amber-500", children: event.agent }),
                  " \u2014 ",
                  event.reason
                ] })
              ] }),
              /* @__PURE__ */ jsxs7("div", { className: "flex items-center gap-4 text-[10px] font-mono text-muted-foreground", children: [
                /* @__PURE__ */ jsxs7("span", { children: [
                  "Confidence: ",
                  (event.confidence * 100).toFixed(0),
                  "%"
                ] }),
                /* @__PURE__ */ jsxs7("span", { children: [
                  "Mode: ",
                  event.mode
                ] })
              ] })
            ] }),
            event.status === "pending" && /* @__PURE__ */ jsxs7("div", { className: "flex gap-2 ml-4 shrink-0", children: [
              /* @__PURE__ */ jsxs7(Button, { size: "sm", onClick: () => handleApprove(event.id), className: "bg-emerald-500 hover:bg-emerald-600 text-black", children: [
                /* @__PURE__ */ jsx18(CheckCircle, { className: "h-3 w-3 mr-1" }),
                "Approve"
              ] }),
              /* @__PURE__ */ jsxs7(Button, { size: "sm", variant: "outline", onClick: () => handleReject(event.id), children: [
                /* @__PURE__ */ jsx18(XCircle, { className: "h-3 w-3 mr-1" }),
                "Reject"
              ] })
            ] }),
            event.status !== "pending" && /* @__PURE__ */ jsx18(
              Badge,
              {
                variant: "outline",
                className: `text-[10px] py-0 ${event.status === "approved" ? "text-emerald-500 border-emerald-500/30 bg-emerald-500/5" : "text-destructive border-destructive/30 bg-destructive/5"}`,
                children: event.status.toUpperCase()
              }
            )
          ] }) }) })
        },
        event.id
      )) }) }) }) })
    ] })
  ] });
}

// src/hooks/useOversight.ts
import { useState as useState5, useCallback } from "react";
function useOversight(initialMode = "checkpoint") {
  const [mode, setMode] = useState5(initialMode);
  const [queue, setQueue] = useState5([]);
  const [history, setHistory] = useState5([]);
  const requestApproval = useCallback(
    async (payload) => {
      if (mode === "shadow") {
        if (payload.confidence >= 0.8 && payload.risk !== "critical") {
          return true;
        }
      }
      if (mode === "recovery") {
        return false;
      }
      return new Promise((resolve) => {
        setQueue((prev) => [...prev, payload]);
        const checkInterval = setInterval(() => {
          setQueue((current) => {
            const item = current.find(
              (q) => q.action === payload.action && q.agent === payload.agent
            );
            if (!item) {
              clearInterval(checkInterval);
              setHistory((h) => [
                ...h,
                { ...payload, status: "approved", timestamp: (/* @__PURE__ */ new Date()).toISOString() }
              ]);
              resolve(true);
              return current;
            }
            return current;
          });
        }, 500);
      });
    },
    [mode]
  );
  const approve = useCallback((action, agent) => {
    setQueue((prev) => {
      const item = prev.find((q) => q.action === action && q.agent === agent);
      if (item) {
        setHistory((h) => [
          ...h,
          { ...item, status: "approved", timestamp: (/* @__PURE__ */ new Date()).toISOString() }
        ]);
      }
      return prev.filter((q) => !(q.action === action && q.agent === agent));
    });
  }, []);
  const reject = useCallback((action, agent) => {
    setQueue((prev) => {
      const item = prev.find((q) => q.action === action && q.agent === agent);
      if (item) {
        setHistory((h) => [
          ...h,
          { ...item, status: "rejected", timestamp: (/* @__PURE__ */ new Date()).toISOString() }
        ]);
      }
      return prev.filter((q) => !(q.action === action && q.agent === agent));
    });
  }, []);
  const state = {
    mode,
    pendingCount: queue.length,
    approvedCount: history.filter((h) => h.status === "approved").length,
    rejectedCount: history.filter((h) => h.status === "rejected").length
  };
  return {
    mode,
    setMode,
    queue,
    history,
    state,
    requestApproval,
    approve,
    reject
  };
}
export {
  AgentFleet,
  Avatar,
  AvatarFallback,
  AvatarImage,
  Badge,
  BrainInterface,
  Button,
  Card,
  CardContent,
  CardDescription,
  CardFooter,
  CardHeader,
  CardTitle,
  Dialog,
  DialogClose,
  DialogContent,
  DialogDescription,
  DialogFooter,
  DialogHeader,
  DialogTitle,
  DialogTrigger,
  GovernancePanel,
  Input,
  Label2 as Label,
  OversightPanel,
  Progress,
  ScrollArea,
  ScrollBar,
  Select,
  SelectContent,
  SelectGroup,
  SelectItem,
  SelectLabel,
  SelectScrollDownButton,
  SelectScrollUpButton,
  SelectSeparator,
  SelectTrigger,
  SelectValue,
  Separator2 as Separator,
  Switch,
  Tabs,
  TabsContent,
  TabsList,
  TabsTrigger,
  Textarea,
  badgeVariants,
  buttonVariants,
  cn,
  useOversight
};
