// @billion-business/venture-ui/dist/components/governance/GovernancePanel.js
'use client';

// src/components/governance/GovernancePanel.tsx
import { useState } from "react";
import { Shield, ShieldAlert, ShieldCheck, Lock, Eye, AlertTriangle, Scale } from "lucide-react";

// src/components/ui/card.tsx
import * as React from "react";

// src/lib/utils.ts
import { clsx } from "clsx";
import { twMerge } from "tailwind-merge";
function cn(...inputs) {
  return twMerge(clsx(inputs));
}

// src/components/ui/card.tsx
import { jsx } from "react/jsx-runtime";
var Card = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      className: cn("rounded-xl border bg-card text-card-foreground shadow", className),
      ...props
    }
  )
);
Card.displayName = "Card";
var CardHeader = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("flex flex-col space-y-1.5 p-6", className), ...props })
);
CardHeader.displayName = "CardHeader";
var CardTitle = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("font-semibold leading-none tracking-tight", className), ...props })
);
CardTitle.displayName = "CardTitle";
var CardDescription = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("text-sm text-muted-foreground", className), ...props })
);
CardDescription.displayName = "CardDescription";
var CardContent = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("p-6 pt-0", className), ...props })
);
CardContent.displayName = "CardContent";
var CardFooter = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("flex items-center p-6 pt-0", className), ...props })
);
CardFooter.displayName = "CardFooter";

// src/components/ui/badge.tsx
import { cva } from "class-variance-authority";
import { jsx as jsx2 } from "react/jsx-runtime";
var badgeVariants = cva(
  "inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  {
    variants: {
      variant: {
        default: "border-transparent bg-primary text-primary-foreground shadow hover:bg-primary/80",
        secondary: "border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80",
        destructive: "border-transparent bg-destructive text-destructive-foreground shadow hover:bg-destructive/80",
        outline: "text-foreground",
        success: "border-transparent bg-emerald-500/10 text-emerald-500 hover:bg-emerald-500/20",
        warning: "border-transparent bg-amber-500/10 text-amber-500 hover:bg-amber-500/20"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
function Badge({ className, variant, ...props }) {
  return /* @__PURE__ */ jsx2("div", { className: cn(badgeVariants({ variant }), className), ...props });
}

// src/components/ui/scroll-area.tsx
import * as React2 from "react";
import * as ScrollAreaPrimitive from "@radix-ui/react-scroll-area";
import { jsx as jsx3, jsxs } from "react/jsx-runtime";
var ScrollArea = React2.forwardRef(({ className, children, ...props }, ref) => /* @__PURE__ */ jsxs(
  ScrollAreaPrimitive.Root,
  {
    ref,
    className: cn("relative overflow-hidden", className),
    ...props,
    children: [
      /* @__PURE__ */ jsx3(ScrollAreaPrimitive.Viewport, { className: "h-full w-full rounded-[inherit]", children }),
      /* @__PURE__ */ jsx3(ScrollBar, {}),
      /* @__PURE__ */ jsx3(ScrollAreaPrimitive.Corner, {})
    ]
  }
));
ScrollArea.displayName = ScrollAreaPrimitive.Root.displayName;
var ScrollBar = React2.forwardRef(({ className, orientation = "vertical", ...props }, ref) => /* @__PURE__ */ jsx3(
  ScrollAreaPrimitive.ScrollAreaScrollbar,
  {
    ref,
    orientation,
    className: cn(
      "flex touch-none select-none transition-colors",
      orientation === "vertical" && "h-full w-2.5 border-l border-l-transparent p-[1px]",
      orientation === "horizontal" && "h-2.5 flex-col border-t border-t-transparent p-[1px]",
      className
    ),
    ...props,
    children: /* @__PURE__ */ jsx3(ScrollAreaPrimitive.ScrollAreaThumb, { className: "relative flex-1 rounded-full bg-border" })
  }
));
ScrollBar.displayName = ScrollAreaPrimitive.ScrollAreaScrollbar.displayName;

// src/components/ui/switch.tsx
import * as React3 from "react";
import * as SwitchPrimitives from "@radix-ui/react-switch";
import { jsx as jsx4 } from "react/jsx-runtime";
var Switch = React3.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx4(
  SwitchPrimitives.Root,
  {
    className: cn(
      "peer inline-flex h-5 w-9 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent shadow-sm transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input",
      className
    ),
    ...props,
    ref,
    children: /* @__PURE__ */ jsx4(
      SwitchPrimitives.Thumb,
      {
        className: cn(
          "pointer-events-none block h-4 w-4 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-4 data-[state=unchecked]:translate-x-0"
        )
      }
    )
  }
));
Switch.displayName = SwitchPrimitives.Root.displayName;

// src/components/governance/GovernancePanel.tsx
import { jsx as jsx5, jsxs as jsxs2 } from "react/jsx-runtime";
var DEFAULT_POLICIES = [
  {
    id: "POL-001",
    name: "Budget Gate",
    description: "Prevents any agent action exceeding $500 without human approval. Applies to all financial operations.",
    status: "active",
    tier: "CRITICAL"
  },
  {
    id: "POL-002",
    name: "Financial Reconciliation",
    description: "All payment and invoice operations must be verified against Stripe before execution.",
    status: "active",
    tier: "CRITICAL"
  },
  {
    id: "POL-003",
    name: "Code Sandbox",
    description: "Generated code must pass dry-run in isolated sandbox before deployment. No direct production writes.",
    status: "active",
    tier: "HIGH"
  },
  {
    id: "POL-004",
    name: "PII Redaction",
    description: "All prompts are scanned for emails, phone numbers, and SSNs. Detected PII is redacted before LLM processing.",
    status: "active",
    tier: "HIGH"
  },
  {
    id: "POL-005",
    name: "Deployment Guard",
    description: "Production deployments require passing all preflight checks: lint, test, security scan, bundle size.",
    status: "active",
    tier: "HIGH"
  },
  {
    id: "POL-006",
    name: "External API Rate Limit",
    description: "Agent scraping and external API calls are rate-limited to prevent abuse and IP bans.",
    status: "active",
    tier: "MEDIUM"
  },
  {
    id: "POL-007",
    name: "Auth Boundary",
    description: "Agents cannot access or modify authentication tokens, session cookies, or user credentials.",
    status: "active",
    tier: "CRITICAL"
  }
];
var ICON_MAP = {
  "POL-001": Lock,
  "POL-002": Scale,
  "POL-003": ShieldCheck,
  "POL-004": Eye,
  "POL-005": ShieldAlert,
  "POL-006": AlertTriangle,
  "POL-007": Lock
};
var TIER_COLORS = {
  CRITICAL: "text-destructive border-destructive/30 bg-destructive/5",
  HIGH: "text-amber-500 border-amber-500/30 bg-amber-500/5",
  MEDIUM: "text-blue-500 border-blue-500/30 bg-blue-500/5",
  LOW: "text-muted-foreground border-border bg-muted/50"
};
function GovernancePanel({ policies: initialPolicies, onToggle }) {
  const [policies, setPolicies] = useState(initialPolicies || DEFAULT_POLICIES);
  const handleToggle = (id, currentStatus) => {
    const newStatus = currentStatus === "active" ? "inactive" : "active";
    setPolicies((prev) => prev.map((p) => p.id === id ? { ...p, status: newStatus } : p));
    onToggle?.(id, newStatus);
  };
  const activeCount = policies.filter((p) => p.status === "active").length;
  const criticalActive = policies.filter((p) => p.tier === "CRITICAL" && p.status === "active").length;
  return /* @__PURE__ */ jsxs2("div", { className: "flex-1 flex flex-col p-6 overflow-hidden", children: [
    /* @__PURE__ */ jsxs2("div", { className: "flex items-center justify-between mb-8", children: [
      /* @__PURE__ */ jsxs2("div", { children: [
        /* @__PURE__ */ jsxs2("h2", { className: "text-xl font-bold flex items-center gap-2", children: [
          /* @__PURE__ */ jsx5(Shield, { className: "h-5 w-5 text-amber-500" }),
          "Agent Governance Engine"
        ] }),
        /* @__PURE__ */ jsx5("p", { className: "text-sm text-muted-foreground mt-1", children: "Machine-enforced boundaries and trust policies." })
      ] }),
      /* @__PURE__ */ jsxs2("div", { className: "flex items-center gap-3", children: [
        /* @__PURE__ */ jsxs2(Badge, { variant: "outline", className: "text-[10px] py-0 border-amber-500/30 text-amber-500 bg-amber-500/5", children: [
          activeCount,
          "/",
          policies.length,
          " ENFORCED"
        ] }),
        criticalActive < policies.filter((p) => p.tier === "CRITICAL").length && /* @__PURE__ */ jsx5(Badge, { variant: "outline", className: "text-[10px] py-0 border-destructive/30 text-destructive bg-destructive/5", children: "CRITICAL GAPS" })
      ] })
    ] }),
    /* @__PURE__ */ jsx5(ScrollArea, { className: "flex-1", children: /* @__PURE__ */ jsx5("div", { className: "grid grid-cols-1 gap-4 pb-8", children: policies.map((policy) => {
      const Icon = ICON_MAP[policy.id] || Shield;
      return /* @__PURE__ */ jsx5(Card, { className: "hover:border-border/80 transition-colors", children: /* @__PURE__ */ jsxs2(CardContent, { className: "p-4 flex items-start justify-between", children: [
        /* @__PURE__ */ jsxs2("div", { className: "flex gap-4", children: [
          /* @__PURE__ */ jsx5("div", { className: "h-10 w-10 rounded-lg bg-muted border flex items-center justify-center shrink-0", children: /* @__PURE__ */ jsx5(Icon, { className: `h-5 w-5 ${policy.status === "active" ? "text-amber-500" : "text-muted-foreground/50"}` }) }),
          /* @__PURE__ */ jsxs2("div", { children: [
            /* @__PURE__ */ jsxs2("div", { className: "flex items-center gap-2 mb-1", children: [
              /* @__PURE__ */ jsx5("h3", { className: "font-bold text-sm", children: policy.name }),
              /* @__PURE__ */ jsx5(Badge, { variant: "outline", className: `text-[10px] py-0 ${TIER_COLORS[policy.tier]}`, children: policy.tier })
            ] }),
            /* @__PURE__ */ jsx5("p", { className: "text-xs text-muted-foreground max-w-md", children: policy.description }),
            /* @__PURE__ */ jsxs2("div", { className: "mt-3 flex items-center gap-4 text-[10px] font-mono text-muted-foreground", children: [
              /* @__PURE__ */ jsxs2("span", { children: [
                "ID: ",
                policy.id
              ] }),
              /* @__PURE__ */ jsxs2("span", { className: "flex items-center gap-1", children: [
                /* @__PURE__ */ jsx5("span", { className: `inline-block w-1.5 h-1.5 rounded-full ${policy.status === "active" ? "bg-green-500" : "bg-red-500"}` }),
                "STATE: ",
                policy.status === "active" ? "ENFORCED" : "BYPASSED"
              ] })
            ] })
          ] })
        ] }),
        /* @__PURE__ */ jsx5("div", { className: "flex flex-col items-end gap-2", children: /* @__PURE__ */ jsx5(Switch, { checked: policy.status === "active", onCheckedChange: () => handleToggle(policy.id, policy.status) }) })
      ] }) }, policy.id);
    }) }) })
  ] });
}
export {
  GovernancePanel
};
