// @billion-business/venture-ui/dist/components/governance/OversightPanel.js
'use client';

// src/components/governance/OversightPanel.tsx
import { useState } from "react";
import { Eye, Shield, AlertTriangle, CheckCircle, XCircle, Clock, Activity } from "lucide-react";

// src/components/ui/card.tsx
import * as React from "react";

// src/lib/utils.ts
import { clsx } from "clsx";
import { twMerge } from "tailwind-merge";
function cn(...inputs) {
  return twMerge(clsx(inputs));
}

// src/components/ui/card.tsx
import { jsx } from "react/jsx-runtime";
var Card = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx(
    "div",
    {
      ref,
      className: cn("rounded-xl border bg-card text-card-foreground shadow", className),
      ...props
    }
  )
);
Card.displayName = "Card";
var CardHeader = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("flex flex-col space-y-1.5 p-6", className), ...props })
);
CardHeader.displayName = "CardHeader";
var CardTitle = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("font-semibold leading-none tracking-tight", className), ...props })
);
CardTitle.displayName = "CardTitle";
var CardDescription = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("text-sm text-muted-foreground", className), ...props })
);
CardDescription.displayName = "CardDescription";
var CardContent = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("p-6 pt-0", className), ...props })
);
CardContent.displayName = "CardContent";
var CardFooter = React.forwardRef(
  ({ className, ...props }, ref) => /* @__PURE__ */ jsx("div", { ref, className: cn("flex items-center p-6 pt-0", className), ...props })
);
CardFooter.displayName = "CardFooter";

// src/components/ui/button.tsx
import * as React2 from "react";
import { Slot } from "@radix-ui/react-slot";
import { cva } from "class-variance-authority";
import { jsx as jsx2 } from "react/jsx-runtime";
var buttonVariants = cva(
  "inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0",
  {
    variants: {
      variant: {
        default: "bg-primary text-primary-foreground shadow hover:bg-primary/90",
        destructive: "bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90",
        outline: "border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground",
        secondary: "bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80",
        ghost: "hover:bg-accent hover:text-accent-foreground",
        link: "text-primary underline-offset-4 hover:underline"
      },
      size: {
        default: "h-9 px-4 py-2",
        sm: "h-8 rounded-md px-3 text-xs",
        lg: "h-10 rounded-md px-8",
        icon: "h-9 w-9"
      }
    },
    defaultVariants: {
      variant: "default",
      size: "default"
    }
  }
);
var Button = React2.forwardRef(
  ({ className, variant, size, asChild = false, ...props }, ref) => {
    const Comp = asChild ? Slot : "button";
    return /* @__PURE__ */ jsx2(
      Comp,
      {
        className: cn(buttonVariants({ variant, size, className })),
        ref,
        ...props
      }
    );
  }
);
Button.displayName = "Button";

// src/components/ui/badge.tsx
import { cva as cva2 } from "class-variance-authority";
import { jsx as jsx3 } from "react/jsx-runtime";
var badgeVariants = cva2(
  "inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2",
  {
    variants: {
      variant: {
        default: "border-transparent bg-primary text-primary-foreground shadow hover:bg-primary/80",
        secondary: "border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80",
        destructive: "border-transparent bg-destructive text-destructive-foreground shadow hover:bg-destructive/80",
        outline: "text-foreground",
        success: "border-transparent bg-emerald-500/10 text-emerald-500 hover:bg-emerald-500/20",
        warning: "border-transparent bg-amber-500/10 text-amber-500 hover:bg-amber-500/20"
      }
    },
    defaultVariants: {
      variant: "default"
    }
  }
);
function Badge({ className, variant, ...props }) {
  return /* @__PURE__ */ jsx3("div", { className: cn(badgeVariants({ variant }), className), ...props });
}

// src/components/ui/scroll-area.tsx
import * as React3 from "react";
import * as ScrollAreaPrimitive from "@radix-ui/react-scroll-area";
import { jsx as jsx4, jsxs } from "react/jsx-runtime";
var ScrollArea = React3.forwardRef(({ className, children, ...props }, ref) => /* @__PURE__ */ jsxs(
  ScrollAreaPrimitive.Root,
  {
    ref,
    className: cn("relative overflow-hidden", className),
    ...props,
    children: [
      /* @__PURE__ */ jsx4(ScrollAreaPrimitive.Viewport, { className: "h-full w-full rounded-[inherit]", children }),
      /* @__PURE__ */ jsx4(ScrollBar, {}),
      /* @__PURE__ */ jsx4(ScrollAreaPrimitive.Corner, {})
    ]
  }
));
ScrollArea.displayName = ScrollAreaPrimitive.Root.displayName;
var ScrollBar = React3.forwardRef(({ className, orientation = "vertical", ...props }, ref) => /* @__PURE__ */ jsx4(
  ScrollAreaPrimitive.ScrollAreaScrollbar,
  {
    ref,
    orientation,
    className: cn(
      "flex touch-none select-none transition-colors",
      orientation === "vertical" && "h-full w-2.5 border-l border-l-transparent p-[1px]",
      orientation === "horizontal" && "h-2.5 flex-col border-t border-t-transparent p-[1px]",
      className
    ),
    ...props,
    children: /* @__PURE__ */ jsx4(ScrollAreaPrimitive.ScrollAreaThumb, { className: "relative flex-1 rounded-full bg-border" })
  }
));
ScrollBar.displayName = ScrollAreaPrimitive.ScrollAreaScrollbar.displayName;

// src/components/governance/OversightPanel.tsx
import { motion, AnimatePresence } from "motion/react";
import { jsx as jsx5, jsxs as jsxs2 } from "react/jsx-runtime";
var DEFAULT_EVENTS = [
  {
    id: "ESC-001",
    agent: "FinanceAuditor_01",
    action: "Process refund of $5,200",
    reason: "Value threshold exceeded ($5K limit)",
    confidence: 0.92,
    timestamp: "18:22:01",
    mode: "checkpoint",
    status: "pending",
    risk: "high"
  },
  {
    id: "ESC-002",
    agent: "ContentDraft_01",
    action: "Publish marketing email to 12,000 contacts",
    reason: "Mass communication requires human review",
    confidence: 0.78,
    timestamp: "18:23:15",
    mode: "shadow",
    status: "pending",
    risk: "medium"
  },
  {
    id: "ESC-003",
    agent: "MarketScraper_01",
    action: "Scrape competitor pricing data",
    reason: "External data collection",
    confidence: 0.95,
    timestamp: "18:24:00",
    mode: "shadow",
    status: "approved",
    risk: "low"
  },
  {
    id: "ESC-004",
    agent: "LeadEnricher_01",
    action: "Update CRM with 340 new leads",
    reason: "Bulk data modification",
    confidence: 0.65,
    timestamp: "18:25:30",
    mode: "checkpoint",
    status: "pending",
    risk: "medium"
  }
];
var RISK_COLORS = {
  low: "text-emerald-500 bg-emerald-500/10 border-emerald-500/30",
  medium: "text-amber-500 bg-amber-500/10 border-amber-500/30",
  high: "text-orange-500 bg-orange-500/10 border-orange-500/30",
  critical: "text-destructive bg-destructive/10 border-destructive/30"
};
var MODE_CONFIG = {
  shadow: { label: "SHADOW", icon: Eye, description: "Agents act unless flagged" },
  checkpoint: { label: "CHECKPOINT", icon: Shield, description: "Agents pause for approval" },
  recovery: { label: "RECOVERY", icon: AlertTriangle, description: "Humans take control" }
};
function OversightPanel({
  events: initialEvents,
  onApprove,
  onReject,
  mode = "checkpoint",
  onModeChange
}) {
  const [events, setEvents] = useState(initialEvents || DEFAULT_EVENTS);
  const pendingCount = events.filter((e) => e.status === "pending").length;
  const approvedCount = events.filter((e) => e.status === "approved").length;
  const rejectedCount = events.filter((e) => e.status === "rejected").length;
  const handleApprove = (id) => {
    setEvents((prev) => prev.map((e) => e.id === id ? { ...e, status: "approved" } : e));
    onApprove?.(id);
  };
  const handleReject = (id) => {
    setEvents((prev) => prev.map((e) => e.id === id ? { ...e, status: "rejected" } : e));
    onReject?.(id);
  };
  const ModeIcon = MODE_CONFIG[mode].icon;
  return /* @__PURE__ */ jsxs2("div", { className: "h-full p-6 space-y-6", children: [
    /* @__PURE__ */ jsxs2("div", { className: "flex items-center justify-between", children: [
      /* @__PURE__ */ jsxs2("div", { children: [
        /* @__PURE__ */ jsxs2("h2", { className: "text-2xl font-black text-amber-500 tracking-tighter flex items-center gap-3 italic", children: [
          /* @__PURE__ */ jsx5(Eye, { className: "h-8 w-8" }),
          "HUMAN OVERSIGHT"
        ] }),
        /* @__PURE__ */ jsx5("p", { className: "text-muted-foreground text-xs font-bold uppercase tracking-widest mt-1", children: "AI supervisor dashboard \u2014 approve, reject, escalate" })
      ] }),
      /* @__PURE__ */ jsx5("div", { className: "flex gap-2", children: Object.keys(MODE_CONFIG).map((m) => {
        const C = MODE_CONFIG[m];
        return /* @__PURE__ */ jsxs2(
          Button,
          {
            variant: mode === m ? "default" : "outline",
            size: "sm",
            onClick: () => onModeChange?.(m),
            className: mode === m ? "bg-amber-500 text-black hover:bg-amber-600" : "",
            children: [
              /* @__PURE__ */ jsx5(C.icon, { className: "h-3 w-3 mr-1" }),
              C.label
            ]
          },
          m
        );
      }) })
    ] }),
    /* @__PURE__ */ jsxs2("div", { className: "grid grid-cols-4 gap-4", children: [
      /* @__PURE__ */ jsx5(Card, { children: /* @__PURE__ */ jsxs2(CardContent, { className: "p-4 flex items-center gap-3", children: [
        /* @__PURE__ */ jsx5(Clock, { className: "h-8 w-8 text-amber-500" }),
        /* @__PURE__ */ jsxs2("div", { children: [
          /* @__PURE__ */ jsx5("p", { className: "text-2xl font-black", children: pendingCount }),
          /* @__PURE__ */ jsx5("p", { className: "text-[10px] text-muted-foreground uppercase font-black", children: "PENDING" })
        ] })
      ] }) }),
      /* @__PURE__ */ jsx5(Card, { children: /* @__PURE__ */ jsxs2(CardContent, { className: "p-4 flex items-center gap-3", children: [
        /* @__PURE__ */ jsx5(CheckCircle, { className: "h-8 w-8 text-emerald-500" }),
        /* @__PURE__ */ jsxs2("div", { children: [
          /* @__PURE__ */ jsx5("p", { className: "text-2xl font-black", children: approvedCount }),
          /* @__PURE__ */ jsx5("p", { className: "text-[10px] text-muted-foreground uppercase font-black", children: "APPROVED" })
        ] })
      ] }) }),
      /* @__PURE__ */ jsx5(Card, { children: /* @__PURE__ */ jsxs2(CardContent, { className: "p-4 flex items-center gap-3", children: [
        /* @__PURE__ */ jsx5(XCircle, { className: "h-8 w-8 text-destructive" }),
        /* @__PURE__ */ jsxs2("div", { children: [
          /* @__PURE__ */ jsx5("p", { className: "text-2xl font-black", children: rejectedCount }),
          /* @__PURE__ */ jsx5("p", { className: "text-[10px] text-muted-foreground uppercase font-black", children: "REJECTED" })
        ] })
      ] }) }),
      /* @__PURE__ */ jsx5(Card, { children: /* @__PURE__ */ jsxs2(CardContent, { className: "p-4 flex items-center gap-3", children: [
        /* @__PURE__ */ jsx5(Activity, { className: "h-8 w-8 text-blue-500" }),
        /* @__PURE__ */ jsxs2("div", { children: [
          /* @__PURE__ */ jsx5("p", { className: "text-2xl font-black", children: events.length }),
          /* @__PURE__ */ jsx5("p", { className: "text-[10px] text-muted-foreground uppercase font-black", children: "TOTAL" })
        ] })
      ] }) })
    ] }),
    /* @__PURE__ */ jsxs2(Card, { children: [
      /* @__PURE__ */ jsx5(CardHeader, { className: "pb-3", children: /* @__PURE__ */ jsxs2("div", { className: "flex items-center justify-between", children: [
        /* @__PURE__ */ jsxs2("div", { children: [
          /* @__PURE__ */ jsxs2(CardTitle, { className: "flex items-center gap-2", children: [
            /* @__PURE__ */ jsx5(ModeIcon, { className: "h-4 w-4 text-amber-500" }),
            MODE_CONFIG[mode].label,
            " MODE"
          ] }),
          /* @__PURE__ */ jsx5(CardDescription, { children: MODE_CONFIG[mode].description })
        ] }),
        /* @__PURE__ */ jsx5(Badge, { variant: "outline", className: "text-[10px] py-0 border-amber-500/30 text-amber-500 bg-amber-500/5", children: MODE_CONFIG[mode].label })
      ] }) }),
      /* @__PURE__ */ jsx5(CardContent, { children: /* @__PURE__ */ jsx5(ScrollArea, { className: "h-[400px] pr-4", children: /* @__PURE__ */ jsx5("div", { className: "space-y-4", children: /* @__PURE__ */ jsx5(AnimatePresence, { children: events.map((event) => /* @__PURE__ */ jsx5(
        motion.div,
        {
          initial: { opacity: 0, y: 10 },
          animate: { opacity: 1, y: 0 },
          exit: { opacity: 0, x: -20 },
          children: /* @__PURE__ */ jsx5(Card, { className: `border ${event.status === "pending" ? "border-amber-500/30" : "border-border/50"}`, children: /* @__PURE__ */ jsx5(CardContent, { className: "p-4", children: /* @__PURE__ */ jsxs2("div", { className: "flex items-start justify-between", children: [
            /* @__PURE__ */ jsxs2("div", { className: "space-y-2 flex-1", children: [
              /* @__PURE__ */ jsxs2("div", { className: "flex items-center gap-2", children: [
                /* @__PURE__ */ jsx5(Badge, { variant: "outline", className: `text-[10px] py-0 ${RISK_COLORS[event.risk]}`, children: event.risk.toUpperCase() }),
                /* @__PURE__ */ jsx5("span", { className: "text-[10px] font-mono text-muted-foreground", children: event.id }),
                /* @__PURE__ */ jsx5("span", { className: "text-[10px] font-mono text-muted-foreground", children: event.timestamp })
              ] }),
              /* @__PURE__ */ jsxs2("div", { children: [
                /* @__PURE__ */ jsx5("p", { className: "text-sm font-bold", children: event.action }),
                /* @__PURE__ */ jsxs2("p", { className: "text-xs text-muted-foreground mt-1", children: [
                  /* @__PURE__ */ jsx5("span", { className: "font-mono text-amber-500", children: event.agent }),
                  " \u2014 ",
                  event.reason
                ] })
              ] }),
              /* @__PURE__ */ jsxs2("div", { className: "flex items-center gap-4 text-[10px] font-mono text-muted-foreground", children: [
                /* @__PURE__ */ jsxs2("span", { children: [
                  "Confidence: ",
                  (event.confidence * 100).toFixed(0),
                  "%"
                ] }),
                /* @__PURE__ */ jsxs2("span", { children: [
                  "Mode: ",
                  event.mode
                ] })
              ] })
            ] }),
            event.status === "pending" && /* @__PURE__ */ jsxs2("div", { className: "flex gap-2 ml-4 shrink-0", children: [
              /* @__PURE__ */ jsxs2(Button, { size: "sm", onClick: () => handleApprove(event.id), className: "bg-emerald-500 hover:bg-emerald-600 text-black", children: [
                /* @__PURE__ */ jsx5(CheckCircle, { className: "h-3 w-3 mr-1" }),
                "Approve"
              ] }),
              /* @__PURE__ */ jsxs2(Button, { size: "sm", variant: "outline", onClick: () => handleReject(event.id), children: [
                /* @__PURE__ */ jsx5(XCircle, { className: "h-3 w-3 mr-1" }),
                "Reject"
              ] })
            ] }),
            event.status !== "pending" && /* @__PURE__ */ jsx5(
              Badge,
              {
                variant: "outline",
                className: `text-[10px] py-0 ${event.status === "approved" ? "text-emerald-500 border-emerald-500/30 bg-emerald-500/5" : "text-destructive border-destructive/30 bg-destructive/5"}`,
                children: event.status.toUpperCase()
              }
            )
          ] }) }) })
        },
        event.id
      )) }) }) }) })
    ] })
  ] });
}
export {
  OversightPanel
};
